---
title: "Database_Wrangle"
author: "Jake Eisaguirre"
date: "2022-09-23"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

if (!require(librarian)){
  install.packages("librarian")
  library(librarian)
}

# librarian downloads, if not already downloaded, and reads in needed packages
librarian::shelf(tidyverse, here, janitor, lubridate, RPostgres, rstudioapi, DBI, parsedate, stringr, hms,
                 anytime, terra, sp, rgdal, RWmisc, proj4, uuid)
```

# create location table (panama, brazil, usa)
```{r}

location_table <- data.frame(c("panama", "brazil", "usa"))

colnames(location_table) <- c("location")

location_table <- location_table %>% 
  group_by(location) %>% 
  mutate(location_id = UUIDgenerate())

write_csv(location_table, here("clean_tables", "location.csv"))


```


# create region table

## manually create USA regions
```{r}

usa_regions <- data.frame(c("pennsylvania", "vermont", "new mexico", "tennessee", "louisiana", "california"),
                          c("usa", "usa", "usa", "usa", "usa", "usa"))

colnames(usa_regions) <- c("region", "location")

```

## read in and clean panama legacy data and pull out regions
```{r}

panama_raw <- read_csv(here("data", "panama_legacy", "panama_data.csv")) %>% 
  clean_names() %>% 
  mutate(frog_presence = ifelse(!nofrog == "No frogs", 1, 0 ),
         date = parse_date(date),
         detection_type = if_else(detection_type == "call", "aural", detection_type)) %>% 
  mutate_at(vars(frog_presence), ~replace_na(., 1)) %>% 
  select(!c(nofrog)) %>% 
  relocate(frog_presence, .before = observers) %>% 
  rename(duration_minutes = surv_length,
         elevation_m = elevation, # double check this is correct
         wind_speed_m_s = wind_speed, # need to double check kph, I see some showing "0.4/s"
         air_temp_c = air_temp, # double check this is correct
         water_temp_c = water_temp, # double check this is correct
         dissolved_o2_percent = dis_o2, # insert units
         conductivity_insertunits = conductivity, #insert units
         tds_insertunits = tds, # insert units
         salinity_small = salinity, # insert units
         cloud_cover_percent = cloud_cover,# double check this is correct
         humidity_percent = humidity, # double check this is correct
         pressure_hg = pressure_in_hg,
         number_observers = num_obs,
         salinity_large = salinity0) %>% 
  mutate_all(str_to_lower) %>% 
  # Data Cleaning
  mutate(swab_id = if_else(sort_id == "5133", "NA", swab_id), # remove swab_id for escaped flotator
         notes_amphib1 = if_else(swab_id == "121022_03", "date and swab_id mismatch ok", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "121022_04", "date and swab_id mismatch ok", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "131212_kc", "date and swab_id mismatch ok", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "131212_kc2", "date and swab_id mismatch ok", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "130706_r01", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "130707_r09", "off transect", notes_amphib1),
         swab_id = if_else(survey_id == "446", str_replace(swab_id, "j", "g"), swab_id),
         end_time = case_when(swab_id == "121112_03" ~ "22:15:00",
                              swab_id == "121112_04" ~ "22:15:00",
                              swab_id == "121112_05" ~ "22:15:00",
                              swab_id == "121112_06" ~ "22:15:00",
                              swab_id == "121112_07" ~ "22:15:00",
                              swab_id == "121112_08" ~ "22:15:00",
                              swab_id == "121112_09" ~ "22:15:00",
                              TRUE ~ as.character(end_time)),
         notes_amphib1 = if_else(swab_id == "130709_r01", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "130709_r02", "off transect", notes_amphib1),
         species = if_else(swab_id == "130709_r01", "pristimantis spp", species),
         species = if_else(swab_id == "130709_r02", "pristimantis spp", species),
         notes_amphib1 = if_else(swab_id == "160709_a05", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "160709_a06", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "160709_a07", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "160709_a08", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "160709_a15", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "160709_a16", "off transect", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "181214_j27", "off transect", notes_amphib1),
         species = if_else(swab_id == "160710_a07", "craugastor gollmeri?", species),
         notes_amphib1 = if_else(sort_id == 137, "at meter 159, counted 75 black tadpoles", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "181215_j04", "caught after end time", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "181215_j05", "caught after end time", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "190801_01", "after transect was completed", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "190801_02", "after transect was completed", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "190801_03", "after transect was completed", notes_amphib1),
         notes_amphib1 = if_else(swab_id == "190801_04", "after transect was completed", notes_amphib1),
         end_time = if_else(sort_id == "5064", "13:34:00", end_time),
         survey_time = if_else(sort_id == "177", "night", survey_time),
         survey_time = if_else(sort_id == "205", "day", survey_time),
         survey_time = if_else(sort_id == "917", "day", survey_time),
         survey_time = if_else(sort_id == "1179", "day", survey_time),
         survey_time = if_else(sort_id == "173", "day", survey_time),
         survey_time = if_else(sort_id == "918", "day", survey_time),
         survey_time = if_else(sort_id == "174", "night", survey_time),
         survey_time = if_else(sort_id == "175", "night", survey_time),
         survey_time = if_else(sort_id == "176", "night", survey_time))



# regions
panama_regions <- panama_raw %>% 
  mutate(region = str_to_lower(region),
         location = "panama") %>% 
  select(region, location) %>% 
  unique()
```

## manually create brazil legacy data region (santa virginia)
```{r}

brazil_regions <- data.frame(c("santa virginia", "boraceia"), c("brazil"))

colnames(brazil_regions) <- c("region", "location")

```

## combine region tables
```{r}

regions <- rbind(brazil_regions, panama_regions, usa_regions) %>% 
  mutate(region = str_replace_all(region, " ", "_")) %>% 
  group_by(region) %>% 
  mutate(region_id = UUIDgenerate(output = c("uuid")),
         location_id = as.UUID(""))

write_csv(regions, here("clean_tables", "regions.csv"))

rm(brazil_regions, panama_regions, usa_regions)
gc()
```


# create site table 

## site table sierra nevada

### conenct sierra nevada database
```{r}
tryCatch({
    drv <- dbDriver("Postgres")
    print("Connecting to Database…")
    connection <- dbConnect(drv,
                 dbname = Sys.getenv("sn_dbname"),
                 host = Sys.getenv("sn_host"),
                 port = Sys.getenv("sn_port"),
                 user = Sys.getenv("sn_user"),
                 password = Sys.getenv("sn_password"),
                 timezone=NULL)
    print("Database Connected!")
    },
    error=function(cond) {
            print("Unable to connect to Database.")
    })

dbExecute(connection, "set search_path to public")
```
### query site table 
```{sql, connection=connection, output.var = "sn_site"}


select s.*
from site s


```

### pull out sn site table
```{r}


sn_site_table <- sn_site %>%
  rename(site = id,
         elevation_m = elevation) %>%
  mutate(region = "california",
        location = "usa",
        utm_zone = if_else(wilderness == "desolation", 10, 11)) %>%
  select(!c(name, drainage, county, jurisdiction)) %>%
  group_by(site) %>%
  unique()

#write_csv(sn_site_table, here("clean_tables", "sn_site_table.csv"))
```


## pull out panama sites
```{r}

panama_sites <- panama_raw %>% 
  mutate(region = str_to_lower(region),
         site = str_to_lower(site),
         location = "panama",
         utm_zone = 17) %>% 
  select(region, site, location, utme, utmn, elevation_m, utm_zone) %>% 
  unique()


```

## pull out serdp sites - TRUE SITE TABLE (still need to convert lat and long to proper UTM Zone)
```{r}

serdp_sites <- read_csv(here("data", "serdp", "serdp_site_table.csv")) %>% 
  clean_names() %>% 
  rename(site_comments = notes,
         region = dod_location) %>% 
  # align serdp site names with current penn site names
  mutate(site = str_to_lower(site),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         location = "usa",
         region = if_else(region == "pa", "pennsylvania", region),
         region = if_else(region == "tn", "tennessee", region),
         region = if_else(region == "vt", "vermont", region),
         region = if_else(region == "nm", "new mexico", region),
         region = if_else(region == "la", "louisiana", region),
         region = if_else(site_code == "la7", "louisiana", region),
         region = if_else(site_code == "la8", "louisiana", region),
         site = if_else(site_code == "nm22", "n. caballo lake 2", site),
         site = if_else(site_code == "nm30", "n. caballo lake 3", site),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site),
         utm_zone = floor((long + 180) / 6) + 1) %>% 
  select(!c( temp_id))


#split data by utm_zone
utmsplit <- split(serdp_sites, serdp_sites$utm_zone)

#apply function to each groupings
dfs <- lapply(utmsplit, function(dx){
   utm_zone <- dx$utm_zone[1]  #get utm zone for the group
   project_string <- paste0("+proj=utm +zone=", utm_zone)
   
   v <- vect(dx, c("long", "lat"), crs="+proj=longlat")
   u <- terra::project(v, project_string)
   utm <- crds(u)

   #add new columns to data frame
   cbind(dx, utm)
})

#combine individuals data frames back into one.
answer <- dplyr::bind_rows(dfs) %>% 
  select(site_code, x, y) %>% 
  rename(utme=x,
         utmn=y)

serdp_sites <- left_join(serdp_sites, answer, by = c("site_code")) %>% 
  select(!c(lat, long))

# t <- serdp_sites %>% 
#   select(site, site_code) %>% 
#   write_csv(here("data", "serdp", "edna", "join_site.csv"))

```

## load penn ribbitr sites
```{r}
penn_sites <- read_csv(here("data", "penn", "penn_sites.csv")) %>% 
  clean_names() %>% 
  select(!c(observer,date)) %>% 
  mutate(region = "pennsylvania",
         utm_zone = 17)
```
## pull out brazil sites
```{r}

brazil_legacy_sites <- read_csv(here("data", "brazil_legacy", "brazil_site.csv")) %>% 
  unite(site, area:environment, sep = " ") %>% 
  rename(lat = latitude_start,
         long = longitude_start) %>% 
  select(!c(latitude_end, longitude_end)) %>% 
  mutate(region = "santa virginia",
         location = "brazil",
         utm_zone = "23")


v <- vect(brazil_legacy_sites, c("long", "lat"), crs="+proj=longlat")
u <- terra::project(v, "+proj=utm +zone=23")
utm <- crds(u)

i <- which(!is.na(brazil_legacy_sites$long))
brazil_legacy_sites[i, c("utme", "utmn")] <- utm

brazil_legacy_sites <- brazil_legacy_sites %>% 
  select(!c(long, lat))

brazil_sites <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  rename(region = region_id,
         site = site_id) %>% 
  select(region, site) %>% 
  mutate(utm_zone = "23") %>% 
  mutate(site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         region = str_to_lower(region),
         region = str_replace_all(region, " ", "_"),
         location = "brazil") %>% 
  group_by(site) %>% 
  unique()

brazil_locs <- read_csv(here("data", "brazil", "survey_site_locs.csv")) %>% 
  rename(latitude = lat,
         longitude = lon) %>% 
  mutate(latitude = as.numeric(latitude),
         longitude = as.numeric(longitude),
         site = str_to_lower(site))

v <- terra::vect(brazil_locs, c("longitude", "latitude"), crs="+proj=longlat")
u <- terra::project(v, "+proj=utm +zone=23")
utm <- terra::crds(u)

i <- which(!is.na(brazil_locs$longitude))
brazil_locs[i, c("utme", "utmn")] <- utm

brazil_sites <- brazil_sites %>% 
  left_join(brazil_locs, by = c("site")) %>% 
  select(!c(longitude, latitude))

final_brazil_sites <- plyr::rbind.fill(brazil_sites, brazil_legacy_sites) 

```

## combine site tables
```{r}

sites <- plyr::rbind.fill(final_brazil_sites, penn_sites, serdp_sites, panama_sites, sn_site_table) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         region = str_replace_all(region, " ", "_"),
         site = str_replace_all(site, "-", "_"),
         site = str_remove(site, "\\.")) %>% 
  group_by(site) %>%
   mutate(across(everything(), ~ .x[order(.x)])) %>%
   slice_head(n = 1) %>%
   ungroup() %>% 
  mutate(utme = round(as.numeric(utme, 4)),
         utmn = round(as.numeric(utmn, 4))) %>% 
  rename(area_sqr_m = area,
         depth_m = depth) %>% 
  relocate(site_comments, .after = wilderness) %>% 
  group_by(site) %>% 
  mutate(site_id = UUIDgenerate(output = c("uuid")),
         region_id = as.UUID(""),
         utme = if_else(site == "rabbit_stream", 618036, utme),
         utmn = if_else(site == "rabbit_stream", 960391, utmn),
         utme = if_else(site == 'admin_pond', 547460.95, utme),
         utmn = if_else(site == 'admin_pond', 4610302.89, utmn),
         utme = if_else(site == "silenciosa", 0545146, utme),
         utmn = if_else(site == "silenciosa", 0958387, utmn),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site))

# brazil_locs <- read_csv(here("data", "brazil", "site_locs.csv")) %>% 
#   rename(latitude = lat,
#          longitude = lon)

# t <- sites %>%
#   filter(region == "pennsylvania")


write_csv(sites, here("clean_tables", "sites.csv"))

#rm(panama_sites, brazil_legacy_sites, penn_sites, sn_site, sn_site_table)
gc()
```

# create visit table

## pull out unique visits for penn1 ribbitr

### load all year 1 penn data
```{r, message=F}

munged_Amphibian_Survey <- read_csv(here("data", "penn", "munged_Amphibian_Survey.csv")) %>% 
  select(location, date, wind_speed_ms, air_temperature_c, water_temperature_c, conductivity_us, ph) %>% 
  rename(site = location) %>% 
  mutate(date = format_iso_8601(date),
         site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_"))

munged_Amphibian_Processing_App <- read_csv(here("data", "penn", "munged_Amphibian_Processing_App.csv"))

munged_Amphibian_Captured_Information <-  read_csv(here("data", "penn","munged_Amphibian_Captured_Information.csv"))
                       
munged_Amphibian_Captured_Information_Repeatable <- read_csv(here("data","penn", "munged_Amphibian_Captured_Information_Repeatable.csv"))

munged_Amphibian_Processing_App_Repeatable <- read_csv(here("data", "penn","munged_Amphibian_Processing_App_Repeatable.csv"))

munged_VisualEncounter_Survey <- read_csv(here("data", "penn","munged_VisualEncounter_Survey.csv"))
  
munged_VisualEncounter_Survey_Repeatable <- read_csv(here("data", "penn","munged_VisualEncounter_Survey_Repeatable.csv"))

munged_Acoustic_Survey <- read_csv(here("data", "penn","munged_Acoustic_Survey.csv"))

munged_Acoustic_Survey_Repeatable <- read_csv(here("data/penn/munged_Acoustic_Survey_Repeatable.csv"))


# IDS in the commetns fixes
cap_comments <- munged_Amphibian_Processing_App_Repeatable %>% 
  slice_head(n = 21) %>% 
  mutate(capture_comments = str_remove(capture_comments, "; red groin"),
         capture_comments = str_remove(capture_comments, "; skin tag on throat"),
         capture_comments = str_remove(capture_comments, "; probably greater than 100 grams, no scale available")) %>% 
  mutate(across(ends_with("_id"), 
        ~replace(.x, is.na(.x), capture_comments)),
        bag_id = "") %>% 
    mutate(across(ends_with("_id_1"), 
        ~replace(.x, is.na(.x), capture_comments)),
        capture_comments = "")


munged_Amphibian_Processing_App_Repeatable <- munged_Amphibian_Processing_App_Repeatable %>% 
  slice_tail(n = -21) %>% 
  rbind(cap_comments)


```

### join different fulcrum app forms per survey

#### capture
```{r}

penn1_capture <- right_join(right_join(munged_Amphibian_Captured_Information,
                                                        munged_Amphibian_Captured_Information_Repeatable, 
                                                        by = c("fulcrum_id" = "fulcrum_parent_id")),
                                            right_join(munged_Amphibian_Processing_App,
                                                       munged_Amphibian_Processing_App_Repeatable, 
                                                       by =c('fulcrum_id' = 'fulcrum_parent_id')),
                                            by = c('location', 'date', 'bag_id')) %>% 
  select(!c( "fulcrum_id.x", "created_at.x.x", "fulcrum_id.y.x", "created_at.y.x", 
            "fulcrum_id.y.y", "created_at.x.y", "fulcrum_id.y.y.y","created_at.y.y", "time_of_capture.y", 
            "body_temperature.y", "microhabitat_type.y","microhabitat_temperature.y", "observer_other", "processor_other",
            "capture_type_other", "sex_other", "species")) %>% 
  rename(survey_comment = survey_comments.x,
         visit_comment = survey_comments.y,
         time_of_capture = time_of_capture.x,
         body_temperature_c = body_temperature.x,
         microhabitat_type = microhabitat_type.x,
         microhabitat_temperature_c = microhabitat_temperature.x,
         site = location) %>% 
  mutate(date = format_iso_8601(date),
         observer = str_to_lower(observer),
         site = str_to_lower(site),
         bag_id = str_to_lower(bag_id),
         microhabitat_type = str_to_lower(microhabitat_type),
         processor = str_to_lower(processor),
         capture_type = str_to_lower(capture_type),
         life_stage = str_to_lower(life_stage),
         species_capture = str_to_lower(species_capture),
         sex = str_to_lower(sex),
         site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_")) %>% 
  unique() %>% 
    mutate(start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night")) %>% 
  select(!c(start_hour, end_hour)) %>% 
  mutate(species_capture = paste(species_capture, species_capture_other, sep = ""),
         species_capture = str_replace(species_capture, "NA", ""),
         region = "pa",
         location = "usa",
         detection_type = "capture") %>% 
  select(!c(species_capture_other))%>% 
  mutate(survey_time = if_else(is.na(survey_time), "night", survey_time),
          duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60) %>%
  relocate(duration_min, .after = end_time) 

penn1_capture$duration_min <- str_sub(penn1_capture$duration_min, -4) %>% 
  as.numeric()

penn1_capture <- penn1_capture %>% 
  left_join(munged_Amphibian_Survey, by = c("site", "date")) %>% 
  rename(photo_id = bag_photo)

june_july_missing_data <- read_csv(here("data", "penn", "2022_missing_data", "clean_missing_june_july_data.csv")) %>% 
  mutate(detection_type = "capture") %>% 
  rename(body_temp_c = body_temperature)

penn1_capture <- plyr::rbind.fill(penn1_capture, june_july_missing_data) %>%
  mutate(amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca4", "2022-05-18-admin-raca4", amp_id_1), #manual data fixes
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca6", "2022-05-18-admin-raca6", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca7", "2022-05-18-admin-raca7", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca9", "2022-05-18-admin-raca9", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca10", "2022-05-18-admin-raca10", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca12", "2022-05-18-admin-raca12", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca14", "2022-05-18-admin-raca14", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca15", "2022-05-18-admin-raca15", amp_id_1),
         # amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-rapi15", "2022-05-18-admin-rapi15", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18‐admin-rapi16", "2022-05-18‐admin-rapi16", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-rapi17", "2022-05-18-admin-rapi17", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-rapi18", "2022-05-18-admin-rapi18", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-pscr19", "2022-05-18-admin-pscr19", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-pscr20", "2022-05-18-admin-pscr20", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-05-19-admin-rapi1", "2022-05-19-admin-rapi1", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-08-19-admin-rapi2", "2022-05-19-admin-rapi2", amp_id_1),
         amp_id_1 = if_else(bd_swab_tube_id == "2022-08-19-admin-rapi3", "2022-05-19-admin-rapi3", amp_id_1),
        amp_id_1 = if_else(bd_swab_tube_id == "2022-05-18-admin-raca1", "2022-05-18-admin-raca1", amp_id_1),
        antibody_id_1 = if_else(bd_swab_tube_id == "2022-05-19-admin-rapi1", "2022-05-19-admin-rapi1", antibody_id_1),
        antibody_id_1 = if_else(bd_swab_tube_id == "2022-08-19-admin-rapi2", "2022-05-19-admin-rapi2", antibody_id_1),
        antibody_id_1 = if_else(bd_swab_tube_id == "2022-08-15-admin-rapi3", "2022-05-19-admin-rapi3", antibody_id_1),
        bd_swab_tube_id = if_else(bd_swab_tube_id == "2022-08-19-admin-rapi2", "2022-05-19-admin-rapi2", bd_swab_tube_id),
        bacterial_swab_tube_id = if_else(bacterial_swab_tube_id == "2022-08-19-admin-rapi2", "2022-05-19-admin-rapi2", bacterial_swab_tube_id),
        mucusome_id = if_else(mucusome_id == "2022-08-19-admin-rapi2", "2022-05-19-admin-rapi2", mucusome_id),
        bd_swab_tube_id = if_else(bd_swab_tube_id == "2022-08-15-admin-rapi3", "2022-05-19-admin-rapi3", bd_swab_tube_id),
        bacterial_swab_tube_id = if_else(bacterial_swab_tube_id == "2022-08-15-admin-rapi3", "2022-05-19-admin-rapi3", bacterial_swab_tube_id),
        mucusome_id = if_else(mucusome_id == "2022-08-15-admin-rapi3", "2022-05-19-admin-rapi3", mucusome_id),
        amp_id_1 = if_else(mucusome_id == "2022-05-19-admin-rapi3", "2022-05-19-admin-rapi3", amp_id_1),
        amp_id_1 = if_else(amp_id_1 == "AMPBath00048", "AMPBath00038", amp_id_1),
        amp_id_1 = if_else(mucusome_id == "MucBath00052", "AMPBath00052", amp_id_1))

```

#### VES
```{r}

penn1_VES <- left_join(munged_VisualEncounter_Survey, munged_VisualEncounter_Survey_Repeatable, 
                       by = c('fulcrum_id' = 'fulcrum_parent_id')) %>% 
  select(observer, date, location, start_time, end_time, species_ves, count_ves, comments_ves) %>%
  unique() %>%
  drop_na(species_ves) %>% 
  mutate(observer = str_to_lower(observer),
         date = format_iso_8601(date),
         location = str_to_lower(location),
         species_ves = str_to_lower(species_ves),
         comments_ves = str_to_lower(comments_ves)) %>% 
  rename(count = count_ves,
         site = location) %>% 
      mutate(start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night"),
         region = "pa",
         location = "usa",
         detection_type = "visual") %>% 
  select(!c(start_hour, end_hour))%>% 
  mutate(survey_time = if_else(is.na(survey_time), "night", survey_time),
         site = if_else(is.na(site), "phelps pond", site),
         duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60)%>%
  relocate(duration_min, .before = species_ves)

penn1_VES$duration_min <- str_sub(penn1_VES$duration_min, -4) %>% 
  as.numeric()

```

#### aural
```{r}
penn1_aural <-left_join(munged_Acoustic_Survey, munged_Acoustic_Survey_Repeatable, by = c('fulcrum_id' = 'fulcrum_parent_id')) %>%
  select(observer, date, location, start_time, end_time, species_acoustic, call_index, acoustic_comments) %>%
  unique() %>%
  mutate(observer = str_to_lower(observer),
         date = format_iso_8601(date),
         location = str_to_lower(location),
         species_acoustic = str_to_lower(species_acoustic),
         call_index = str_to_lower(call_index),
         acoustic_comments = str_to_lower(acoustic_comments)) %>% 
  rename(site = location) %>% 
  mutate(duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60,
         detection_type = "acoustic",
          region = "pa",
         location = "usa",
         start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night")) %>%
  relocate(duration_min, .before = species_acoustic)

penn1_aural$duration_min <- str_sub(penn1_aural$duration_min, -4) %>% 
  as.numeric()

rm(munged_Amphibian_Processing_App, munged_Amphibian_Captured_Information,munged_Amphibian_Captured_Information_Repeatable,
   munged_Amphibian_Processing_App_Repeatable,munged_VisualEncounter_Survey,munged_VisualEncounter_Survey_Repeatable,
   munged_Acoustic_Survey,munged_Acoustic_Survey_Repeatable)
gc()
```

### pull out visits based on date and survey_time(night/day) for VES, aural, and capture
```{r}
penn1_VES_visit <- penn1_VES %>% 
  select(date, survey_time, site, detection_type)

penn1_capture_visit <- penn1_capture %>% 
  select(date, survey_time, site, detection_type)

penn1_aural_visit <- penn1_aural %>% 
  select(date, survey_time, site, detection_type)
```

### bind all penn1 visits and filter for duplicated uniqueness based on date, survey_time, and site to get true penn1 visit table
```{r}

penn_visit_table <- plyr::rbind.fill(penn1_aural_visit, penn1_VES_visit, penn1_capture_visit) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id(),
         site = str_replace(site, "-", "_"),
         site = str_replace_all(site, " ", "_"),
         date = format_iso_8601(date)) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(detection_type, temp_id))

#rm(penn1_aural_visit, penn1_VES_visit, penn1_capture_visit)
gc()
```

## pull out uinque sn visits

### query data
```{sql, connection=connection, output.var = "sn_visit"}


select s.id as site, v.* 
from site s
join visit v on s.id = v.site_id;


```

### clean up sn_visit
```{r}

sn_visit_table <- sn_visit %>% 
  select(!c(id, site_id)) %>% 
  rename(visit_notes = comment,
         date = visit_date) %>% 
  mutate(survey_time = "day") %>% 
  group_by(site, date, survey_time) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

#write_csv(sn_visit_table, here("clean_tables", "sn_visit.csv"))

#rm(sn_visit)
gc()
```


## read in serdp unique visits - need to either read in all data and format sites as above or adjust in different script. Either method is fine since data will not be changing
```{r}

serdp_visit_table <- read_csv(here("data", "serdp", "serdp_visit_table.csv")) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace(site, "-", "_"),
         site = str_remove(site, "\\."),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         site = str_replace_all(site, " ", "_"),
         survey_time = str_to_lower(survey_time),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site)) %>% 
  rename(visit_notes = notes) %>% 
  group_by(site, date, survey_time) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

```

## read panama data and pull out unique visits
```{r}
panama_visit_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "visit.csv")) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

panama_visit_table <- panama_raw %>% 
  select(site, date, survey_time) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "[[:space:]]", "")) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id)) %>% 
  plyr::rbind.fill(panama_visit_2022)


```

## read in brazil data unique visits
```{r}

brazil_legacy_visit_table <- read_csv(here("data", "brazil_legacy", "brazil_visit.csv")) %>% 
  mutate(site = str_replace_all(site, "[[:space:]]", ""))

brazil_visit_table <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  select(date, site_id, survey_time) %>% 
  rename(site = site_id) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace(site, " ", "_")) %>% 
  group_by(site, date, survey_time) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!temp_id)

brazil_visit <- rbind(brazil_visit_table, brazil_legacy_visit_table)
  

```

## bind all visit tables
```{r}

visit_table <- plyr::rbind.fill(brazil_visit, panama_visit_table,
                                penn_visit_table, serdp_visit_table, sn_visit_table) %>% 
  mutate(date = format_iso_8601(date),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site)) %>% 
  rename(visit_comments = visit_notes) %>% 
  relocate(visit_comments, .after = visit_status) %>% 
  relocate(date, .before = campaign) %>% 
  relocate(survey_time, .after = date) %>% 
  group_by(site, date, survey_time) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id)) %>% 
  mutate(visit_id = UUIDgenerate(output = c("uuid")),
         site_id = as.UUID(""))

write_csv(visit_table, here("clean_tables", "visit.csv"))

# rm(brazil_legacy_visit_table, panama_visit_table, penn_visit_table, serdp_visit_table,
#    sn_visit_table)
gc()



# a <- sites %>% 
#   select(site) %>% 
#   unique()
# 
# b <- visit_table %>% 
#   select(site) %>% 
#   unique()
# 
# # sites never visites #12
# z <- anti_join(a, b, by = c("site")) 

```

# create survey tables for each region/dataset?

## brazil legacy survey table
```{r}

brazil_legacy_survey_table <- read_csv(here("data", "brazil_legacy", "brazil_survey.csv")) %>% 
  mutate(detection_type = "capture", 
         site = str_replace_all(site, "[[:space:]]", ""),
         date = format_iso_8601(date),
         duration_minutes = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_minutes = duration_minutes/60)

brazil_legacy_survey_table$duration_minutes <- str_sub(brazil_legacy_survey_table$duration_minutes, -4) %>% 
  as.numeric()

new_brazil_survey <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  rename(site = site_id) %>% 
  select(site, date, survey_time, observer) %>% 
  group_by(site, date, survey_time) %>% 
  unique() %>% 
  mutate(date = format_iso_8601(date),
         site = str_to_lower(site),
         detection_type = "capture")

brazil_survey_ves <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  filter(region_id == "Boraceia") %>% 
  rename(site = site_id) %>% 
  select(site, date, survey_time, observer) %>% 
  group_by(site, date, survey_time) %>% 
  unique() %>% 
  mutate(date = format_iso_8601(date),
         site = str_to_lower(site),
         detection_type = "visual")

brazil_survey <- plyr::rbind.fill(brazil_legacy_survey_table, new_brazil_survey, brazil_survey_ves) %>% 
  mutate(start_time = as.character(start_time),
         end_time = as.character(end_time))
  

write_csv(brazil_survey, here("clean_tables", "brazil_legacy_survey.csv"))
```

## penn survey table
```{r}

penn_survey_table <- plyr::rbind.fill(penn1_aural, penn1_VES, penn1_capture) %>% 
  select(c(site, date, observer, detection_type, start_time, end_time, 
           duration_min, detection_type, survey_time, wind_speed_ms, air_temperature_c, 
           water_temperature_c, ph, total_dissolved_solids, survey_comments)) %>% 
  group_by(date, site, detection_type, survey_time) %>% 
  mutate(temp_reg_id = cur_group_id()) %>% 
  filter(!duplicated(temp_reg_id)) %>% 
  select(!c(temp_reg_id)) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_"),
         detection_type = if_else(detection_type == "acoustic", "aural", detection_type),
         date = format_iso_8601(date),
         start_time = as.character(start_time),
         end_time = as.character(end_time)) %>% 
  rename(duration_minutes = duration_min,
         air_temp_c = air_temperature_c,
         water_temp_c = water_temperature_c,
         p_h = ph,
         tds_ppm = total_dissolved_solids,
         wind_speed_m_s = wind_speed_ms) %>% 
  relocate(observer, .after = duration_minutes)

write_csv(penn_survey_table, here("clean_tables", "penn_survey.csv"))
  

```

## panama survey table
```{r}

panama_survey_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "survey.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  group_by(date, site, survey_time, detection_type) %>% 
  mutate(new_survey_id = cur_group_id(),
         detection_type = if_else(detection_type == "call", "aural", detection_type)) %>% 
  filter(!duplicated(new_survey_id)) %>% 
  select(!c(new_survey_id, e_dna)) %>% 
  rename(wind_speed_m_s = wind_speed_ms)

panama_survey_table <- panama_raw %>% 
  select(c((1:5), (6:15), (21:35), salinity_large, detection_type)) %>% 
  select(!c(survey_id, sort_id, merge_id, region, salinity_large, e_dna)) %>% 
  mutate(site = str_replace_all(site, " ", "_")) %>% 
  group_by(date, site, survey_time, detection_type) %>% 
  mutate(new_survey_id = cur_group_id()) %>% 
  filter(!duplicated(new_survey_id)) %>% 
  select(!c(new_survey_id)) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  plyr::rbind.fill(panama_survey_2022) %>% 
  rename(conductivity_uS_cm = conductivity_insertunits,
         salinity_ppt = salinity_small,
         tds_ppm = tds_insertunits,
         survey_comments = notes_survey,
         relative_humidty_percent = humidity_percent) %>% 
  mutate(dissolved_o2_percent = as.numeric(dissolved_o2_percent), 
         salinity_ppt = if_else(salinity_ppt > "1", "", salinity_ppt),
         salinity_ppt = na_if(salinity_ppt, ""),
         dissolved_o2_percent = case_when(dissolved_o2_percent > 100 ~ 100,
                                          dissolved_o2_percent < 100 ~ dissolved_o2_percent),
         wind_speed_m_s = if_else(wind_speed_m_s == "23.7", "", wind_speed_m_s),
         wind_speed_m_s = na_if(wind_speed_m_s, ""),
        wind_speed_m_s = case_when(str_detect(wind_speed_m_s, "ph", negate = TRUE)~ wind_speed_m_s),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "/s", ""),
         wind_speed_m_s = str_replace_all(wind_speed_m_s, ", gust to 3.7", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "kt", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "k/hr", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, " ", ""),
        transect = str_to_lower(str_replace_all(transect, " ", "_")),
        transect = str_replace_all(transect, ",", "_"),
        transect = str_replace_all(transect, "-", "_"),
        pressure_hg = (as.numeric(pressure_hg) * 0.4911),
        survey_time = if_else(survey_time == "-9999", "day", survey_time)) %>% 
  relocate(start_time, .before = transect) %>% 
  relocate(end_time, .after = start_time) %>% 
  relocate(survey_comments, .after = soil_humidity_m3m3) %>% 
  relocate(duration_minutes, .after = end_time) %>% 
  rename(pressure_psi = pressure_hg,
         observer = observers)



write_csv(panama_survey_table, here("clean_tables", "panama_survey.csv"))
```

## read in serdp survey table
```{r}

serdp_survey_table <- read_csv(here("data", "serdp", "serdp_survey_table.csv")) %>% 
  mutate(survey_notes = str_to_lower(survey_notes),
         precipitation_last_48_h = str_to_lower(precipitation_last_48_h),
         survey_notes = str_replace_all(survey_notes, "[^[:alnum:]]", " "),
         weather_condition_notes = str_replace_all(weather_condition_notes, "[^[:alnum:]]", " "),
         site = str_to_lower(site),
         site = str_replace(site, "-", "_"),
         site = str_remove(site, "\\."),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         site = str_replace_all(site, " ", "_"),
         survey_time = str_to_lower(survey_time),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site),
         detection_type = "capture",
         date = format_iso_8601(date),
         duration_minutes = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_minutes = duration_minutes/60,
         start_time = as.character(start_time),
         end_time = as.character(end_time)) %>% 
  unite("wind_speed_mpers_min", c("wind_speed_mpers_min", "wind_speed_min_mpers"), na.rm = T) %>% 
  unite("wind_speed_mpers_max", c("wind_speed_mpers_max", "wind_speed_max_mpers"), na.rm = T) %>% 
  unite("pressure_hg", c("pressure_in_hg", "pressure_in"), na.rm = T) %>% 
  select(!c(pressure_mb, e_dna_sample, pressure_hg)) %>% 
  mutate(percent_vegetation_cover = if_else(percent_vegetation_cover == "huge lake", "", percent_vegetation_cover),
         wind_speed_mpers = wind_speed_mpers * 0.447,
         wind_speed_mpers_max = as.numeric(wind_speed_mpers_max) * 0.447,
         wind_speed_mpers_min = as.numeric(wind_speed_mpers_min) * 0.447) %>% 
  #mutate_all(na_if, "") %>% 
  relocate(start_time, .before = site_code) %>% 
  relocate(end_time, .after = start_time) %>% 
  relocate(duration_minutes, .after = end_time) %>% 
  rename(survey_comments = survey_notes,
         conductivity_uS_cm = conductivity,
         tds_ppm = total_dissolved_solids,
         air_temp_c = air_temp_deg_c,
         water_temp_c = water_temp_deg_c,
         air_temp_c_drop = air_temp_deg_c_drop,
         relative_humidity_percent = relative_humidity,
         relative_humidity_drop_percent = relative_humidity_drop,
         number_observers = no_of_investigators,
         wind_speed_max_m_s = wind_speed_mpers_max,
         wind_speed_min_m_s = wind_speed_mpers_min,
         wind_speed_m_s = wind_speed_mpers,
         observer = investigator_i_ds)%>% 
  group_by(site, date, survey_time, detection_type) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

serdp_survey_table$duration_minutes <- str_sub(serdp_survey_table$duration_minutes, -4) %>% 
  as.numeric()

write_csv(serdp_survey_table, here("clean_tables", "serdp_survey.csv"))
```

## sn survey table

### query data
```{sql, connection=connection, output.var = "sn_survey"}


select v.site_id as site, v.visit_date, s.* 
from visit v
join survey s on s.visit_id = v.id;


```

### clean sn survey table - need to figure out what to do about CMR and Swab
```{r}

sn_survey_table <- sn_survey %>%
  select(!c(id, visit_id)) %>% 
  rename(detection_type = survey_type,
         date = visit_date,
         survey_comments = comment) %>% 
  mutate(survey_time = "day",
         detection_type = as.character(detection_type),
         detection_type = if_else(detection_type == "cmr", "capture", detection_type),
         detection_type = if_else(detection_type == "swab", "capture", detection_type),
         site = as.character(site),
         date = format_iso_8601(date),
         start_time = as.character(start_time),
         end_time = as.character(end_time)) %>% 
  group_by(site, date, survey_time, detection_type) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id)) %>% 
  rename(duration_minutes = duration,
         air_temp_c = air_temp,
         water_temp_c = water_temp) %>% 
  relocate(c(start_time:duration_minutes), .before = wind)
  
write_csv(sn_survey_table, here("clean_tables", "sn_survey_table.csv"))

#rm(sn_survey)
gc()
```

## bind all survey tables
```{r}

survey <- plyr::rbind.fill(brazil_survey, penn_survey_table, sn_survey_table, panama_survey_table, serdp_survey_table) %>% 
  group_by(site, date, survey_time, detection_type) %>% 
  mutate(survey_id = UUIDgenerate(output = c("uuid")),
         visit_id = as.UUID(""),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site)) %>% 
  select(!c(conductivity_uS_cm, qaqc_survey, frog_presence))

write_csv(survey, here("clean_tables", "survey.csv"))

```


# create VES table

## penn1 ves table
```{r}

penn_ves <- penn1_VES %>% 
  select(!c(region, location, observer)) %>% 
  mutate(duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60,
         date = format_iso_8601(date)) %>% 
  relocate(duration_min, .before = species_ves) %>% 
  rename(ves_notes = comments_ves,
         duration_minutes = duration_min)

penn_ves$duration_minutes <- str_sub(penn_ves$duration_minutes, -4) %>% 
  as.numeric()


rm(penn1_VES)
gc()
```

## panama ves table 
```{r}

panama_ves_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "ves.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  rename(ves_notes = notes_aural)

panama_ves <- panama_raw %>% 
  filter(detection_type == "visual") %>% 
  select(site, date, survey_time, start_time, end_time, 
         duration_minutes, species, detection_type,
         quantity, capture_loc, microhab, life_stage, sex, notes_amphib1, 
         notes_amphib2, microhab0) %>% 
  unite(ves_notes, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         count = quantity,
         species_ves = species,
         detection_location = capture_loc) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  plyr::rbind.fill(panama_ves_2022)

  
```

## sn ves table

### query data
```{sql, connection=connection, output.var = "sn_ves"}


select v.site_id as site, v.visit_date as date, s.survey_type as detection_type, vs.*
from visit v
join survey s on v.id = s.visit_id
join visual_survey vs on s.id = vs.survey_id;


```

### clean sn ves data - need to align with other ves still
```{r}

sn_ves_table <- sn_ves %>% 
  select(!c(id, survey_id)) %>% 
  mutate(survey_time = "day",
         species = case_when(species == "hyre" ~ "hyliola_regilla",
                             species == "thel" ~ "thamnophis_elegans",
                             species == "ramu" ~ "rana_muscosa",
                             species == "buca" ~ "anaxyrus_canorus",
                             species == "bubo" ~ "anaxyrus_boreas",
                             species == "thco" ~ "thamnophis_couchii",
                             species == "tato" ~ "taricha_torosa",
                             species == "raca" ~ "rana_catesbiana",
                             species == "clma" ~ "actinemys_marmorata",
                             species == "amma" ~ "ambystoma_macrodactylum",
                             species == "thsi" ~ "thamnophis_sirtalis")) %>% 
  rename(species_ves = species,
         ves_notes = comment,
         microhab = location,
         life_stage = visual_life_stage)


#write_csv(sn_ves_table, here("clean_tables", "sn_ves_table.csv"))

#rm(sn_ves)
gc()
```

## brazil ves data
```{r}
brazil_ves <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  filter(region_id == "Boraceia") %>% 
  select(site_id, date, survey_time, detection_type, species_capture, life_stage, sex, capture_comments) %>% 
  rename(site = site_id,
         ves_notes = capture_comments,
         species_ves = species_capture) %>% 
  mutate(detection_type = "visual",
         site = str_to_lower(site),
         species_ves = str_to_lower(species_ves),
         count = 1)
```



## bind ves tables
```{r}

ves <- plyr::rbind.fill(panama_ves, penn_ves, sn_ves_table, brazil_ves) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         sex = if_else(sex == "unkonwn", "unknown", sex),
         species_ves = str_replace_all(species_ves, " ", "_")) %>% 
  rename(ves_comments = ves_notes) %>% 
  filter(!species_ves == "control") %>% 
  select(!c(start_time, end_time, duration_minutes)) %>% 
  mutate(count = if_else(count == "many", "100", count)) %>% # need to find better fix to "many"
  #group_by() %>% 
  mutate(ves_id = UUIDgenerate(output = "uuid", n = n()),
         survey_id = as.UUID(""),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site))

write_csv(ves, here("clean_tables", "ves.csv"))

#rm(panama_ves, penn_ves, sn_ves_table)
gc()
```


# create aural table

## penn1 aural table
```{r}

penn_aural <- penn1_aural %>% 
  select(!c(region, observer, location, start_hour, end_hour, start_time, end_time, duration_min)) %>% 
  mutate(detection_type = "aural",
         species_acoustic = str_replace_all(species_acoustic, " ", "_"),
         site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_")) %>% 
  rename(species_aural = species_acoustic,
         notes_aural = acoustic_comments)

#rm(penn1_aural)
gc()
```

## panama aural table
```{r}

panama_aural_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "aural.csv")) %>% 
  mutate(date = format_iso_8601(date))

panama_aural <- panama_raw %>% 
  filter(detection_type == "aural") %>% 
  select(site, date, survey_time, 
         species, detection_type,
         quantity, capture_loc, microhab, life_stage, sex, notes_amphib1, 
         notes_amphib2, microhab0) %>% 
  unite(notes_aural, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         count = quantity,
         species_aural = species,
         detection_location = capture_loc) %>% 
  mutate(date = format_iso_8601(date),
         microhab = if_else(is.na(microhab), "unknown", microhab),
         count = if_else(is.na(count), "1", count), #DOUBLE CHECK THIS IS CORRECT
         site = str_replace_all(site, " ", "_"),
         species_aural = str_replace_all(species_aural, " ", "_")) %>% 
  plyr::rbind.fill(panama_aural_2022)

```

## bind aural table
```{r}

aural <- plyr::rbind.fill(penn_aural, panama_aural) %>% 
  mutate(count = if_else(count == "many", "100", count),
         sex = if_else(sex == "unkonwn", "unknown", sex)) %>% # need to find better fix for "many"
  rename(aural_comments = notes_aural) %>% 
  mutate(aural_id = UUIDgenerate(output = c("uuid"), n = n()),
         survey_id = as.UUID(""),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site))
  

write_csv(aural, here("clean_tables", "aural.csv"))

#rm(penn_aural, panama_aural)
gc()
```

# Genomic data

## SERDP Legacy genomic - BD
need to add before capture table since serdp bd genomic data was not present in initial serdp data and need to back log genomic ids
## read in serdp bd genomic data
```{r}

raw_serdp_bd_genom <- read_csv(here("data", "serdp", "bd_genomic", "bd_genomic_serd.csv")) %>% 
  clean_names()

```


## pull out important columns from serdp bd genomic data for db
```{r}

serdp_bd_genom <- raw_serdp_bd_genom %>% 
  select(sample_id, seq_run, c(n_seqs_187:lineage_dapc_2_99cut)) %>% 
  rename(genetic_id = sample_id) %>% 
  filter(!genetic_id %in% c("180223-01C-LA02-LICA", "180223-04D-LA02-LICA", "180223-01F-LA02-LICA",
            "180223-04A-LA02-LICA", "180223-01B-LA02-LISP", "180223-01D-LA02-LISP"))

write_csv(serdp_bd_genom, here("clean_tables", "serdp_bd_genom.csv"))

# used for adding sample_id into serdp table
t <- serdp_bd_genom %>% 
  select(genetic_id) %>% 
  mutate(sample_id=genetic_id)


```


# 2019 Panama Bd data
```{r}
pan_2019_bd <- read_csv(here("qpcr_data", "2019_panama_bd.csv")) %>% 
  #select(!c(site, Region_code, Site_code, SPECIES)) %>% 
  clean_names() %>% 
  rename(bd_swab_id = sample_name) %>% 
  mutate(bd_swab_id = str_to_lower(bd_swab_id),
         bd_ct_mean = if_else(bd_pos_neg == "Negative", 0, bd_ct_mean),
         bd_quantity_mean = if_else(bd_pos_neg == "Negative", 0, bd_quantity_mean))

write_csv(pan_2019_bd, here("clean_tables", "pan_2019_bd.csv"))

```


# Microbiome data

## Newt serdp data - contains ID for microbiome and mucosome and bacterial column...I think
need to add before capture table since serdp newt data is not accurate for binary microbiome column. Missing some data points
```{r}

newt <- read_csv(here("data", "serdp", "newt", "microbiome_antifungal_TTX_mucosome.csv")) %>% 
  mutate(mucosome_id = swab_id,
         microbiome_swab_id = swab_id)

newt <- newt[rowSums(is.na(newt)) != ncol(newt[,2:27]), ]

write_csv(newt, here("clean_tables", "serdp_newt_microbiome_mucosome_antifungal.csv"))

# use for back logging 62 missing IDs
all_newt_ids <- newt %>% 
  select(microbiome_swab_id, mucosome_id, swab_id)

```



# Capture table

## penn1 capture
```{r}

penn_capture <- penn1_capture %>% 
  select(!c(region, location, visit_comment, survey_comment, start_time, end_time,
            observer, duration_min, wind_speed_ms, air_temperature_c, 
           water_temperature_c, ph, total_dissolved_solids, survey_comments, conductivity_us)) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "-", "_"),
         species_capture = str_replace_all(species_capture, " ", "_")) %>% 
         # bd_swab_tube_id = if_else(bd_swab_tube_id == "BdSwab00000", "NA", bd_swab_tube_id),
         # dry_swab_tube_id=if_else(dry_swab_tube_id == "DrySwab00000", "NA", dry_swab_tube_id),
         # crispr_swab_tube_id=if_else(crispr_swab_tube_id == "CRSwab00000", "NA", crispr_swab_tube_id),
         # bacterial_swab_tube_id=if_else(bacterial_swab_tube_id == "BacSwab00000", "NA", bacterial_swab_tube_id),
         # mucosome_id=if_else(mucusome_id == "MucBath00000", "NA", mucusome_id),
         # amp_id_1=if_else(amp_id_1 == "AMPBath00000", "NA", amp_id_1),
         # amp_id_2=if_else(amp_id_2 == "AMPBath00000", "NA", amp_id_2),
         # amp_id_3=if_else(amp_id_3 == "AMPBath00000", "NA", amp_id_3),
         # amp_id_4=if_else(amp_id_4 == "AMPBath00000", "NA", amp_id_4),
         # antibody_id_1=if_else(antibody_id_1 == "AntiBod00000", "NA", antibody_id_1),
         # antibody_id_2=if_else(antibody_id_2 == "AntiBod00000", "NA", antibody_id_2),
         # antibody_id_3=if_else(antibody_id_3 == "AntiBod00000", "NA", antibody_id_3),
         # antibody_id_4=if_else(antibody_id_4 == "AntiBod00000", "NA", antibody_id_4),
         # toe_clip_tube_id=if_else(toe_clip_tube_id == "ToeClip00000", "NA", toe_clip_tube_id)) %>% 
  unite(body_temp_c, c(body_temperature_c, body_temp_c), sep = "", na.rm = T) %>% 
  rename(bag_mass_g = bag_mass,
         svl_mm = snout_vent_length,
         body_and_bag_mass_g = body_and_bag_mass,
         body_mass_g = body_mass,
         substrate_temp_c = microhabitat_temperature_c,
         bacterial_swab_id =  bacterial_swab_tube_id, #glycerol/culture
         microbiome_swab_id = bd_swab_tube_id, #sequence/dry swab
         genetic_id = toe_clip_tube_id,
         amp_id = amp_id_1,
         #body_temp_c = body_temperature_c,
         bd_swab_id = dry_swab_tube_id,
         crispr_id = crispr_swab_tube_id,
         antibody_id = antibody_id_1,
         mucosome_id = mucusome_id) %>% 
  unite(life_stage, c(life_stage, life_stage_other), sep = ",", na.rm = T) %>% 
  mutate(date = format_iso_8601(date))
  

#rm(penn1_capture, penn1_VES, penn1_aural)
gc()
```

## sn capture

### queruy data
```{sql, connection=connection, output.var = "sn_capture"}


select v.site_id as site, v.visit_date as date, s.survey_type as detection_type, cs.*
from visit v
join survey s on v.id = s.visit_id
join capture_survey cs on s.id = cs.survey_id;


```

### clean sn_data 
```{r}

sn_capture_table <- sn_capture %>% 
  select(!c(survey_id, id, pit_tag_ref, tag_new)) %>% 
  mutate(survey_time = "day",
         detection_type = as.character(detection_type),
         detection_type = if_else(detection_type == "cmr", "capture", detection_type),
         detection_type = if_else(detection_type == "swab", "capture", detection_type)) %>% 
  rename(notes_capture = comment,
         body_mass_g = weight,
         svl_mm = length,
         life_stage = capture_life_stage,
         species_capture = species,
         microhabitat_type = location,
         processor = surveyor_id) %>% 
  mutate(microbiome_swab_id = if_else(str_detect(swab_id, "AJS"), swab_id, NA), #metagenomic sequencing so micro
         bacterial_swab_id = if_else(str_detect(swab_id, "BII"), swab_id, NA), #culture/glycerol
         mucosome_id = if_else(str_detect(swab_id, "BII"), swab_id, NA),
         species_capture = case_when(species_capture == "ramu" ~ "rana_muscosa",
                                     species_capture == "hyre" ~ "hyliola_regilla",
                                     species_capture == "bubo" ~ "anaxyrus_boreas",
                                     species_capture == "buca" ~ "anaxyrus_canorus",
                                     species_capture == "tato" ~ "taricha_torosa")) %>% 
  rename(bd_swab_id = swab_id,
         capture_utme = utme,
         capture_utmn = utmn,
         capture_comments = notes_capture)

  
#write_csv(sn_capture_table, here("clean_tables", "sn_capture_table.csv"))
```


## brazil_legacy capture
```{r}
brazil_legacy_capture <- read_csv(here("data", "brazil_legacy", "brazil_capture.csv")) %>% 
  select(!c(17:20)) %>% 
  select(!c(count, family)) %>% 
  rename(species_capture = scientific_name,
         capture_comments = comments,
         bd_swab_id = swab_id,
         bacterial_swab_id = glycerol, #culture/glycerol
         genetic_id = tissue) %>% 
  unite(site, area:environment, sep = "_") %>% 
  mutate(detection_type = "capture",
         site = str_replace(site, " ", "_"),
         site = str_remove(site, " "),
         species_capture = str_replace_all(species_capture, " ", "_"),
         bacterial_swab_id = if_else(bacterial_swab_id == "TRUE", bd_swab_id, NA),
         genetic_id = if_else(genetic_id == "TRUE", bd_swab_id, NA),
         time_of_capture = if_else(time_of_capture == "NA:00", "", time_of_capture))

v <- vect(brazil_legacy_capture, c("capture_longitude", "capture_latitude"), crs="+proj=longlat")
u <- terra::project(v, "+proj=utm +zone=23")
utm <- crds(u)

i <- which(!is.na(brazil_legacy_capture$capture_longitude))
brazil_legacy_capture[i, c("capture_utme", "capture_utmn")] <- utm

brazil_legacy_capture <- brazil_legacy_capture %>% 
  select(!c(capture_longitude, capture_latitude))

new_brazil_capture <- read_csv(here("data", "brazil", "brazil.csv")) %>% 
  clean_names() %>% 
  rename(site = site_id) %>% 
  select(site, date, survey_time, c(species_capture:capture_comments)) %>% 
  mutate(capture_type = "new",
         site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         species_capture = str_to_lower(species_capture),
         detection_type = "capture") %>% 
  rename(bacterial_swab_id =  bacterial_swab_tube_id, #glycerol/culture
         microbiome_swab_id = dry_swab_tube_id, #sequence/dry swab
         svl_mm = snout_vent_length,
         genetic_id = toe_clip_tube_id,
         amp_id = amp_id_1,
         bd_swab_id = bd_swab_tube_id,
         antibody_id = antibody_id_1)

brazil_capture <- plyr::rbind.fill(brazil_legacy_capture, new_brazil_capture)
 
```

## serdp capture table
```{r}

serdp_capture <- read_csv(here("data", "serdp", "serdp_capture_table.csv")) %>% 
  rename(bag_mass_g = mass_of_bag_g,
         time_of_capture = time,
         body_and_bag_mass_g = mass_of_animal_in_bag,
         body_mass_g = mass_of_animal_g,
         body_temp_c = body_temp,
         microhabitat_type = microhabitat_category,
         location_serdp = location) %>% 
  select(!c(species_code, frog_number, frog_id)) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         bd_swab = if_else(bd_swab == 1, sample_id, NA),
         mucosome = if_else(mucosome == 1, sample_id, NA),
         bacterial_swab_id = if_else(microbiome == 1, sample_id, NA), #culture/glycerol bd
         microbiome = if_else(microbiome == 1, sample_id, NA), #sequce/dry swab
         am_ps = if_else(am_ps == 1, sample_id, NA),
         antibodies = if_else(antibodies == 1, sample_id, NA)) %>% 
  rename(bd_swab_id = bd_swab,
         mucosome_id = mucosome,
         amp_id = am_ps,
         microbiome_swab_id = microbiome, #micriobiome column both types, culture and sequence
         substrate_temp_c = substrate_temp,
         antibody_id = antibodies) %>% 
  left_join(t, by = c("sample_id")) %>% 
  select(!c(sample_id))


# back log missing 62 mucosome and microbiome ids
back_log_microb_muc <- serdp_capture %>%
  filter(!is.na(microbiome_swab_id)) %>% 
  select(microbiome_swab_id, mucosome_id, bd_swab_id) %>% 
  filter(bd_swab_id %in% all_newt_ids$swab_id)

Ids_to_back_log <- anti_join(all_newt_ids, back_log_microb_muc, by = c("swab_id" = "bd_swab_id"))

serdp_capture <- left_join(serdp_capture, Ids_to_back_log, by = c("bd_swab_id"="swab_id")) %>% 
  unite(microbiome_swab_id, c("microbiome_swab_id.x", "microbiome_swab_id.y"), na.rm = T) %>% 
  unite(mucosome_id, c("mucosome_id.x", "mucosome_id.y"), na.rm = T)


```

## panama capture table
```{r}

panama_capture_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "capture.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  rename(capture_animal_state = dead,
         bd_swab_id = bdswab,
         mucosome_id = mucosome_water,
         bacterial_swab_id = glycerolswab, #culture bd
         genetic_id = oralswab,
         amp_id = amps,
         microbiome_swab_id = bacteriaswab,
         antibody_id = ampshcl) %>% # dry swab for sequencing 
  mutate(capture_animal_state = case_when(capture_animal_state == "1" ~ "dead",
                                          capture_animal_state == "0" ~ "alive"),
         bd_swab_id = if_else(bd_swab_id == "1", swab_id, NA),
         mucosome_id = if_else(mucosome_id == "1", swab_id, NA),
         bacterial_swab_id = if_else(bacterial_swab_id == "1", swab_id, NA), #culture/glycerol 
         genetic_id = if_else(genetic_id == "1", swab_id, NA),
         amp_id = if_else(amp_id == "1", swab_id, NA),
         antibody_id = if_else(antibody_id == "1", swab_id, NA),
         microbiome_swab_id = if_else(microbiome_swab_id == "1", swab_id, NA)) %>% 
  select(!c(swab_id, count))


# panama data fixes
panama_rabbit_stream_fix <- panama_raw %>% 
  filter(site == "rabbit stream",
         date == "2014-05-27") %>% 
  group_by(swab_id) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!temp_id)


panama_capture <- panama_raw %>%
  filter(!swab_id %in% panama_rabbit_stream_fix$swab_id) %>% 
  rbind(panama_rabbit_stream_fix) %>% 
  filter(detection_type == "capture") %>% 
  select(c(site, date, survey_time, swab_id, species, detection_type, capture_time,
           capture_loc, microhab, body_temp, subs_temp, svl, frog_mass, life_stage, sex, dead, #infected, pcr,
           bdswab, gen_samp, bacteria_swab, am_ps, photo, photo_id, notes_amphib1,
           notes_amphib2, microhab0, pcr, e_dna)) %>% 
  unite(notes_capture, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         time_of_capture = capture_time,
         capture_trx_loc = capture_loc,
         microhabitat_type = microhab,
         body_temp_c = body_temp,
         substrate_temp_c = subs_temp,
         svl_mm = svl,
         body_mass_g = frog_mass) %>% 
  mutate(bdswab = if_else(bdswab=="1", swab_id, NA),
         am_ps = if_else(am_ps=="1", swab_id, NA),
         bacteria_swab = if_else(bacteria_swab=="1", swab_id, NA),
         e_dna = if_else(e_dna=="1", swab_id, NA),
         gen_samp = case_when(gen_samp == "toeclip" ~ swab_id,
                              gen_samp == "buccal" ~ swab_id,
                              gen_samp == "1" ~ swab_id),
         site = str_replace_all(site, " ", "_"),
         species = str_replace_all(species, " ", "_"),
         date = format_iso_8601(date)) %>% 
  rename(amp_id = am_ps,
         genetic_id = gen_samp,
         bd_swab_id = bdswab,
         microbiome_swab_id = bacteria_swab, #dry swab frozen
         species_capture = species,
         capture_animal_state = dead,
         capture_comments = notes_capture) %>% 
  mutate(capture_animal_state = case_when(capture_animal_state == "1" ~ "dead",
                                          capture_animal_state == "0" ~ "alive")) %>% 
  select(!c(swab_id, e_dna)) %>% 
  plyr::rbind.fill(panama_capture_2022) 
  

write_csv(panama_capture, here("data", "panama_legacy", "clean_tables", "clean_pan_cap.csv"))
```

## bind all capture tables
```{r}

capture <- plyr::rbind.fill(panama_capture, brazil_capture, penn_capture, serdp_capture,
                            sn_capture_table) %>% 
  mutate(capture_comments = str_replace_all(capture_comments, "[[:punct:]]", ""),
         site = str_replace_all(site, "-", "_"),
         site = str_replace_all(site, " ", "_"),
         microhabitat_notes = str_replace_all(microhabitat_notes, "[[:punct:]]", ""),
         site = if_else(site == "wood_lab", "wood_lab_pond", site),
         detection_type = "capture") %>% 
  #mutate_all(na_if, "") %>% 
  #relocate(c(bd_swab_id:e_dna), .after = tad_stage) %>% 
  relocate(c(mucosome_id:antibody_id), .after = amp_id) %>% 
  relocate(c(crispr_id:antibody_id_4), .after = antibody_id) %>% 
  relocate(capture_comments, .after = antibody_id_4) %>% 
  mutate(capture_id = UUIDgenerate(output = c("uuid"), n = n()),
         survey_id = as.UUID(""),
         site = if_else(site == "shelbourne_bay", "shelburne_bay", site),
         site = if_else(site == "shelbourne_pond", "shelburne_pond", site)) %>% 
  select(!c(sex_other, amp_id_3, amp_id_4, antibody_id_3, antibody_id_4))

write_csv(capture, here("clean_tables", "capture.csv"))
  

#rm(panama_capture, brazil_legacy_capture, penn_capture, serdp_capture)
gc()
```

# AMP data

## SERPD AMP - contains ID for amp column and antibody column - PA
```{r}

serdp_amp_pa <- read_csv(here("data", "serdp", "amp", "SERDP_AMP.csv")) %>% 
  clean_names() %>% 
  select(!c(date, site, dod_location, entry, day_night, dod_location_2, site_code, frog_id_x, frog_number, species_code,
            time, microhabitat_notes, microhabitat_category, svl_mm, tail_length_mm, mass_of_animal_in_bag, mass_of_bag_g,
            mass_of_animal_g, bd_swab, mucosome, microbiome, am_ps, antibodies, body_temp, substrate_temp, sex, life_stage,
            field_notes, number_of_mites, c(49:101), month, day, year_2)) %>% 
  rename(amp_id = swab_id) %>% 
  mutate(censor = if_else(is.na(censor), "N", censor),
         date_gia = parse_date(date_gia))

#write_csv(serdp_amp_pa, here("clean_tables", "serdp_amp_pa.csv"))

```

## SERDP AMP - TN
```{r}

serdp_amp_tn <- read_csv(here("data", "serdp", "amp", "serdpAMP_TN.csv")) %>% 
  clean_names() %>% 
  select(!c(date:microhabitat_category)) %>% 
  select(!c(svl_mm:life_stage)) %>% 
  select(!c(c(number_of_mites:notes_y), clearcut, buckets, temp_gun_2, flir_2, sample_id_old, year, month, day,
            location_y, x_y)) %>% 
  rename(gia_final_reading = final_reading,
         amp_id = swab_id,
         gia_plate_pos_growth = plate_pos) %>% 
  mutate(censor = if_else(is.na(censor), "N", censor),
         date_gia = parse_date(date_gia))

```

## combine serdp amp data
```{r}

serdp_amp <- plyr::rbind.fill(serdp_amp_pa, serdp_amp_tn)

write_csv(serdp_amp, here("clean_tables", "serdp_amp.csv"))

```


# legacy hobo data

## hobo site & region table
```{r}

hobo_site <- serdp_sites  %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "-", "_")) %>% 
  group_by(site) %>% 
  mutate(hobo_site_id = UUIDgenerate(output = c("uuid")),
         hobo_region_id = as.UUID("")) %>% 
  ungroup()

write_csv(hobo_site, here("clean_tables", "hobo_site.csv"))

hobo_region <- hobo_site %>% 
  select(region) %>% 
  mutate(location = "usa") %>% 
  group_by(region) %>% 
  unique() %>% 
  mutate(hobo_region_id = UUIDgenerate(output = c("uuid")),
         hobo_location_id = as.UUID(""))

write_csv(hobo_region, here("clean_tables", "hobo_region.csv"))

hobo_location <- hobo_site %>% 
  select(location) %>% 
  group_by(location) %>% 
  unique() %>% 
  mutate(hobo_location_id = UUIDgenerate(output = c("uuid")))

write_csv(hobo_location, here("clean_tables", "hobo_location.csv"))

```


## read in serdp hobo data

### shade hobo
```{r}

raw_shade_hobo <- read.csv(here("data", "serdp", "hobo", "Shade_HOBO_2021_01_28.csv")) %>% 
  clean_names() 

shade_hobo <- raw_shade_hobo %>% 
  select(!c(x_1, x, location)) %>% 
  rename(shade_temperature_c = temperature,
         shade_relative_humidity = rh,
         site_code = site,
         #enviro_location = location,
         shade_dew_point_c = dew_point) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))),
         shade_dew_point_c = if_else(shade_dew_point_c == "Logged", "", shade_dew_point_c)) 

#write_csv(shade_hobo, here("clean_tables", "shade_hobo.csv"))

rm(raw_shade_hobo)
gc()
```

### soil hobo
```{r}

raw_soil_hobo <- read_csv(here("data", "serdp", "hobo", "Soil_HOBO_2021_01_28.csv")) %>% 
  clean_names()

soil_hobo <- raw_soil_hobo %>% 
  select(!c(x1, x, location)) %>% 
  rename(soil_temperature_c = temperature,
         site_code = site) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))))
#write_csv(soil_hobo, here("clean_tables", "soil_hobo.csv"))

rm(raw_soil_hobo)
gc()
```

### sun hobo
```{r}

raw_sun_hobo <- read_csv(here("data", "serdp", "hobo", "Sun_HOBO_2021_01_28.csv")) %>% 
  clean_names()

sun_hobo <- raw_sun_hobo %>% 
  select(!c(x1, x, location)) %>% 
  rename(sun_temperature_c = temperature,
         site_code = site,
         #enviro_location = location,
         sun_light_lux = light) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))))

#write_csv(sun_hobo, here("clean_tables", "sun_hobo.csv"))

rm(raw_sun_hobo)
gc()
```

### water hobo

```{r}
raw_water_hobo <- read_csv(here("data", "serdp", "hobo", "Water_HOBO_2021_01_28.csv")) %>% 
  clean_names()

water_hobo <- raw_water_hobo %>% 
  select(!c(x1, x, location)) %>% 
  rename(water_temperature_c = temperature,
         site_code = site,
         #enviro_location = location,
         water_light_lux = light) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))),
         water_light_lux = if_else(water_light_lux == "Logged", "", water_light_lux))

#write_csv(water_hobo, here("clean_tables", "water_hobo.csv"))

rm(raw_water_hobo)
gc()
```

## bind hobos 
```{r}

all_hobos <- left_join(shade_hobo, soil_hobo, by = c("site_code", "date_time")) %>% 
  left_join(sun_hobo, by = c("site_code", "date_time")) %>% 
  left_join(water_hobo, by = c("site_code", "date_time")) %>% 
  select(!c(time_zone.y, time_zone.x.x, time_zone.y.y)) %>% 
  rename(time_zone = time_zone.x) %>% 
  mutate(hobo_id = UUIDgenerate(output = c("uuid"), n = n()),
         hobo_site_id = as.UUID(""))

write_csv(all_hobos, here("clean_tables", "hobo.csv"))  

```


