---
title: "Database_Wrangle"
author: "Jake Eisaguirre"
date: "2022-09-23"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

if (!require(librarian)){
  install.packages("librarian")
  library(librarian)
}

# librarian downloads, if not already downloaded, and reads in needed packages
librarian::shelf(tidyverse, here, janitor, lubridate, RPostgres, rstudioapi, DBI, parsedate, stringr, hms,
                 anytime, terra, sp, rgdal, RWmisc, proj4)
```

# create location table (panama, brazil, usa)
```{r}

location_table <- data.frame(c("panama", "brazil", "usa"))

colnames(location_table) <- c("location")

write_csv(location_table, here("clean_tables", "location.csv"))


```


# create region table

## manually create USA regions
```{r}

usa_regions <- data.frame(c("pennsylvania", "vermont", "new mexico", "tennessee", "louisiana", "california"),
                          c("usa", "usa", "usa", "usa", "usa", "usa"))

colnames(usa_regions) <- c("region", "location")

```

## read in and clean panama legacy data and pull out regions
```{r}

panama_raw <- read_csv(here("data", "panama_legacy", "panama_data.csv")) %>% 
  clean_names() %>% 
  mutate(frog_presence = ifelse(!nofrog == "No frogs", 1, 0 ),
         date = parse_date(date),
         detection_type = if_else(detection_type == "call", "aural", detection_type)) %>% 
  mutate_at(vars(frog_presence), ~replace_na(., 1)) %>% 
  select(!c(nofrog)) %>% 
  relocate(frog_presence, .before = observers) %>% 
  rename(duration_minutes = surv_length,
         elevation_m = elevation, # double check this is correct
         wind_speed_m_s = wind_speed, # need to double check kph, I see some showing "0.4/s"
         air_temp_c = air_temp, # double check this is correct
         water_temp_c = water_temp, # double check this is correct
         dissolved_o2_percent = dis_o2, # insert units
         conductivity_insertunits = conductivity, #insert units
         tds_insertunits = tds, # insert units
         salinity_small = salinity, # insert units
         cloud_cover_percent = cloud_cover,# double check this is correct
         humidity_percent = humidity, # double check this is correct
         pressure_hg = pressure_in_hg,
         number_observers = num_obs,
         salinity_large = salinity0) %>% 
  mutate_all(str_to_lower) 



# regions
panama_regions <- panama_raw %>% 
  mutate(region = str_to_lower(region),
         location = "panama") %>% 
  select(region, location) %>% 
  unique()
```

## manually create brazil legacy data region (santa virginia)
```{r}

brazil_regions <- data.frame(c("santa virginia"), c("brazil"))

colnames(brazil_regions) <- c("region", "location")

```

## combine region tables
```{r}

regions <- rbind(brazil_regions, panama_regions, usa_regions) %>% 
  mutate(region = str_replace_all(region, " ", "_"))

write_csv(regions, here("clean_tables", "regions.csv"))

rm(brazil_regions, panama_regions, usa_regions)
gc()
```


# create site table 

## site table sierra nevada

### conenct sierra nevada database
```{r}
tryCatch({
    drv <- dbDriver("Postgres")
    print("Connecting to Databaseâ€¦")
    connection <- dbConnect(drv,
                 dbname = Sys.getenv("sn_dbname"),
                 host = Sys.getenv("sn_host"),
                 port = Sys.getenv("sn_port"),
                 user = Sys.getenv("sn_user"),
                 password = Sys.getenv("sn_password"),
                 timezone=NULL)
    print("Database Connected!")
    },
    error=function(cond) {
            print("Unable to connect to Database.")
    })

dbExecute(connection, "set search_path to public")
```
### query site table 
```{sql, connection=connection, output.var = "sn_site"}


select s.*
from site s


```

### pull out sn site table
```{r}


sn_site_table <- sn_site %>%
  rename(site = id,
         elevation_m = elevation) %>%
  mutate(region = "california",
        location = "usa",
        utm_zone = 11) %>%
  select(!c(name, drainage, county, jurisdiction)) %>%
  group_by(site) %>%
  unique()

#write_csv(sn_site_table, here("clean_tables", "sn_site_table.csv"))
```


## pull out panama sites
```{r}

panama_sites <- panama_raw %>% 
  mutate(region = str_to_lower(region),
         site = str_to_lower(site),
         location = "panama",
         utm_zone = 17) %>% 
  select(region, site, location, utme, utmn, elevation_m, utm_zone) %>% 
  unique()


```

## pull out serdp sites - TRUE SITE TABLE (still need to convert lat and long to proper UTM Zone)
```{r}

serdp_sites <- read_csv(here("data", "serdp", "serdp_site_table.csv")) %>% 
  clean_names() %>% 
  rename(site_comments = notes,
         region = dod_location) %>% 
  # align serdp site names with current penn site names
  mutate(site = str_to_lower(site),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         location = "usa",
         region = if_else(region == "pa", "pennsylvania", region),
         region = if_else(region == "tn", "tennessee", region),
         region = if_else(region == "vt", "vermont", region),
         region = if_else(region == "nm", "new mexico", region),
         region = if_else(region == "la", "louisiana", region),
         region = if_else(site_code == "la7", "louisiana", region),
         region = if_else(site_code == "la8", "louisiana", region),
         site = if_else(site_code == "nm22", "n. caballo lake 2", site),
         site = if_else(site_code == "nm30", "n. caballo lake 3", site),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site),
         utm_zone = floor((long + 180) / 6) + 1) %>% 
  select(!c( temp_id))


#split data by utm_zone
utmsplit <- split(serdp_sites, serdp_sites$utm_zone)

#apply function to each groupings
dfs <- lapply(utmsplit, function(dx){
   utm_zone <- dx$utm_zone[1]  #get utm zone for the group
   project_string <- paste0("+proj=utm +zone=", utm_zone)
   
   v <- vect(dx, c("long", "lat"), crs="+proj=longlat")
   u <- terra::project(v, project_string)
   utm <- crds(u)

   #add new columns to data frame
   cbind(dx, utm)
})

#combine individuals data frames back into one.
answer <- dplyr::bind_rows(dfs) %>% 
  select(site_code, x, y) %>% 
  rename(utme=x,
         utmn=y)

serdp_sites <- left_join(serdp_sites, answer, by = c("site_code")) %>% 
  select(!c(lat, long))

```

## load penn ribbitr sites
```{r}
penn_sites <- read_csv(here("data", "penn", "penn_sites.csv")) %>% 
  clean_names() %>% 
  select(!c(observer,date)) %>% 
  mutate(region = "pennsylvania",
         utm_zone = 17)
```
## pull out brazil legacy sites
```{r}

brazil_legacy_sites <- read_csv(here("data", "brazil_legacy", "brazil_site.csv")) %>% 
  unite(site, area:environment, sep = " ") %>% 
  rename(lat = latitude_start,
         long = longitude_start) %>% 
  select(!c(latitude_end, longitude_end)) %>% 
  mutate(region = "santa virginia",
         location = "brazil",
         utm_zone = "23")


v <- vect(brazil_legacy_sites, c("long", "lat"), crs="+proj=longlat")
u <- terra::project(v, "+proj=utm +zone=23")
utm <- crds(u)

i <- which(!is.na(brazil_legacy_sites$long))
brazil_legacy_sites[i, c("utme", "utmn")] <- utm

brazil_legacy_sites <- brazil_legacy_sites %>% 
  select(!c(long, lat))

```

## combine site tables
```{r}

sites <- plyr::rbind.fill(brazil_legacy_sites, penn_sites, serdp_sites, panama_sites, sn_site_table) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         region = str_replace_all(region, " ", "_"),
         site = str_replace_all(site, "-", "_"),
         site = str_remove(site, "\\.")) %>% 
  group_by(site) %>%
   mutate(across(everything(), ~ .x[order(.x)])) %>%
   slice_head(n = 1) %>%
   ungroup() %>% 
  mutate(utme = round(as.numeric(utme, 4)),
         utmn = round(as.numeric(utmn, 4))) %>% 
  rename(area_sqr_m = area,
         depth_m = depth)
  
write_csv(sites, here("clean_tables", "sites.csv"))

rm(panama_sites, brazil_legacy_sites, penn_sites, sn_site, sn_site_table)
gc()
```

# create visit table

## pull out unique visits for penn1 ribbitr

### load all year 1 penn data
```{r, message=F}
munged_Amphibian_Processing_App <- read_csv(here("data", "penn", "munged_Amphibian_Processing_App.csv"))

munged_Amphibian_Captured_Information <-  read_csv(here("data", "penn","munged_Amphibian_Captured_Information.csv"))
                       
munged_Amphibian_Captured_Information_Repeatable <- read_csv(here("data","penn", "munged_Amphibian_Captured_Information_Repeatable.csv"))

munged_Amphibian_Processing_App_Repeatable <- read_csv(here("data", "penn","munged_Amphibian_Processing_App_Repeatable.csv"))

munged_VisualEncounter_Survey <- read_csv(here("data", "penn","munged_VisualEncounter_Survey.csv"))
  
munged_VisualEncounter_Survey_Repeatable <- read_csv(here("data", "penn","munged_VisualEncounter_Survey_Repeatable.csv"))

munged_Acoustic_Survey <- read_csv(here("data", "penn","munged_Acoustic_Survey.csv"))

munged_Acoustic_Survey_Repeatable <- read_csv(here("data", "penn","munged_Acoustic_Survey_Repeatable.csv"))


```

### join different fulcrum app forms per survey

#### capture
```{r}

penn1_capture <- right_join(right_join(munged_Amphibian_Captured_Information,
                                                        munged_Amphibian_Captured_Information_Repeatable, 
                                                        by = c("fulcrum_id" = "fulcrum_parent_id")),
                                            right_join(munged_Amphibian_Processing_App,
                                                       munged_Amphibian_Processing_App_Repeatable, 
                                                       by =c('fulcrum_id' = 'fulcrum_parent_id')),
                                            by = c('location', 'date', 'bag_id')) %>% 
  select(!c( "fulcrum_id.x", "created_at.x.x", "fulcrum_id.y.x", "created_at.y.x", 
            "fulcrum_id.y.y", "created_at.x.y", "fulcrum_id.y.y.y","created_at.y.y", "time_of_capture.y", 
            "body_temperature.y", "microhabitat_type.y","microhabitat_temperature.y", "observer_other", "processor_other",
            "capture_type_other", "sex_other", "species")) %>% 
  rename(survey_comment = survey_comments.x,
         visit_comment = survey_comments.y,
         time_of_capture = time_of_capture.x,
         body_temperature_c = body_temperature.x,
         microhabitat_type = microhabitat_type.x,
         microhabitat_temperature_c = microhabitat_temperature.x,
         site = location) %>% 
  mutate(date = format_iso_8601(date),
         observer = str_to_lower(observer),
         site = str_to_lower(site),
         bag_id = str_to_lower(bag_id),
         microhabitat_type = str_to_lower(microhabitat_type),
         processor = str_to_lower(processor),
         capture_type = str_to_lower(capture_type),
         life_stage = str_to_lower(life_stage),
         species_capture = str_to_lower(species_capture),
         sex = str_to_lower(sex)) %>% 
  unique() %>% 
    mutate(start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night")) %>% 
  select(!c(start_hour, end_hour)) %>% 
  mutate(species_capture = paste(species_capture, species_capture_other, sep = ""),
         species_capture = str_replace(species_capture, "NA", ""),
         region = "pa",
         location = "usa",
         detection_type = "capture") %>% 
  select(!c(species_capture_other))%>% 
  mutate(survey_time = if_else(is.na(survey_time), "night", survey_time))



```

#### VES
```{r}

penn1_VES <- left_join(munged_VisualEncounter_Survey, munged_VisualEncounter_Survey_Repeatable, 
                       by = c('fulcrum_id' = 'fulcrum_parent_id')) %>% 
  select(observer, date, location, start_time, end_time, species_ves, count_ves, comments_ves) %>%
  unique() %>%
  drop_na(species_ves) %>% 
  mutate(observer = str_to_lower(observer),
         date = format_iso_8601(date),
         location = str_to_lower(location),
         species_ves = str_to_lower(species_ves),
         comments_ves = str_to_lower(comments_ves)) %>% 
  rename(count = count_ves,
         site = location) %>% 
      mutate(start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night"),
         region = "pa",
         location = "usa",
         detection_type = "visual") %>% 
  select(!c(start_hour, end_hour))%>% 
  mutate(survey_time = if_else(is.na(survey_time), "night", survey_time),
         site = if_else(is.na(site), "phelps pond", site))


```

#### aural
```{r}
penn1_aural <-left_join(munged_Acoustic_Survey, munged_Acoustic_Survey_Repeatable, by = c('fulcrum_id' = 'fulcrum_parent_id')) %>%
  select(observer, date, location, start_time, end_time, species_acoustic, call_index, acoustic_comments) %>%
  unique() %>%
  mutate(observer = str_to_lower(observer),
         date = format_iso_8601(date),
         location = str_to_lower(location),
         species_acoustic = str_to_lower(species_acoustic),
         call_index = str_to_lower(call_index),
         acoustic_comments = str_to_lower(acoustic_comments)) %>% 
  rename(site = location) %>% 
  mutate(duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60,
         detection_type = "acoustic",
          region = "pa",
         location = "usa",
         start_hour = hour(start_time),
         end_hour = hour(end_time),
         survey_time = case_when(start_hour >= 6 & end_hour >= 6 & end_hour < 19 ~ "day", 
                          start_hour >= 19 &  (end_hour < 6 | end_hour <= 23) |
                         (start_hour < 6 & end_hour < 6)~ "night",
                         start_hour >=19 ~"night")) %>%
  relocate(duration_min, .before = species_acoustic)

penn1_aural$duration_min <- str_sub(penn1_aural$duration_min, -4) %>% 
  as.numeric()

rm(munged_Amphibian_Processing_App, munged_Amphibian_Captured_Information,munged_Amphibian_Captured_Information_Repeatable,
   munged_Amphibian_Processing_App_Repeatable,munged_VisualEncounter_Survey,munged_VisualEncounter_Survey_Repeatable,
   munged_Acoustic_Survey,munged_Acoustic_Survey_Repeatable)
gc()
```

### pull out visits based on date and survey_time(night/day) for VES, aural, and capture
```{r}
penn1_VES_visit <- penn1_VES %>% 
  select(date, survey_time, site, detection_type)

penn1_capture_visit <- penn1_capture %>% 
  select(date, survey_time, site, detection_type)

penn1_aural_visit <- penn1_aural %>% 
  select(date, survey_time, site, detection_type)
```

### bind all penn1 visits and filter for duplicated uniqueness based on date, survey_time, and site to get true penn1 visit table
```{r}

penn_visit_table <- plyr::rbind.fill(penn1_aural_visit, penn1_VES_visit, penn1_capture_visit) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id(),
         site = str_replace(site, "-", "_"),
         site = str_replace_all(site, " ", "_"),
         date = format_iso_8601(date)) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(detection_type, temp_id))

rm(penn1_aural_visit, penn1_VES_visit, penn1_capture_visit)
gc()
```

## pull out uinque sn visits

### query data
```{sql, connection=connection, output.var = "sn_visit"}


select s.id as site, v.* 
from site s
join visit v on s.id = v.site_id;


```

### clean up sn_visit
```{r}

sn_visit_table <- sn_visit %>% 
  select(!c(id, site_id)) %>% 
  rename(visit_notes = comment,
         date = visit_date) %>% 
  mutate(survey_time = "day") %>% 
  group_by(site, date, survey_time) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

#write_csv(sn_visit_table, here("clean_tables", "sn_visit.csv"))

rm(sn_visit)
gc()
```


## read in serdp unique visits - need to either read in all data and format sites as above or adjust in different script. Either method is fine since data will not be changing
```{r}

serdp_visit_table <- read_csv(here("data", "serdp", "serdp_visit_table.csv")) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace(site, "-", "_"),
         site = str_remove(site, "\\."),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         site = str_replace_all(site, " ", "_"),
         survey_time = str_to_lower(survey_time),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site)) %>% 
  rename(visit_notes = notes)

```

## read panama data and pull out unique visits
```{r}
panama_visit_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "visit.csv")) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))

panama_visit_table <- panama_raw %>% 
  select(site, date, survey_time) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "[[:space:]]", "")) %>% 
  group_by(date, survey_time, site) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id)) %>% 
  plyr::rbind.fill(panama_visit_2022)


```

## read in brazil legacy data unique visits
```{r}

brazil_legacy_visit_table <- read_csv(here("data", "brazil_legacy", "brazil_visit.csv")) %>% 
  mutate(site = str_replace_all(site, "[[:space:]]", ""))

```

## bind all visit tables
```{r}

visit_table <- plyr::rbind.fill(brazil_legacy_visit_table, panama_visit_table,
                                penn_visit_table, serdp_visit_table, sn_visit_table) %>% 
  mutate(date = format_iso_8601(date))

write_csv(visit_table, here("clean_tables", "visit.csv"))

rm(brazil_legacy_visit_table, panama_visit_table, penn_visit_table, serdp_visit_table,
   sn_visit_table)
gc()

```

# create survey tables for each region/dataset?

## brazil legacy survey table
```{r}

brazil_legacy_survey_table <- read_csv(here("data", "brazil_legacy", "brazil_survey.csv")) %>% 
  mutate(detection_type = "capture", 
         site = str_replace_all(site, "[[:space:]]", ""),
         date = format_iso_8601(date),
         duration_minutes = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_minutes = duration_minutes/60)

brazil_legacy_survey_table$duration_minutes <- str_sub(brazil_legacy_survey_table$duration_minutes, -4) %>% 
  as.numeric()

write_csv(brazil_legacy_survey_table, here("clean_tables", "brazil_legacy_survey.csv"))
```

## penn survey table
```{r}

penn_survey_table <- plyr::rbind.fill(penn1_aural, penn1_VES, penn1_capture) %>% 
  select(c(site, date, observer, detection_type, start_time, end_time, 
           duration_min, detection_type, survey_time)) %>% 
  group_by(date, site, detection_type, survey_time) %>% 
  mutate(temp_reg_id = cur_group_id()) %>% 
  filter(!duplicated(temp_reg_id)) %>% 
  select(!c(temp_reg_id)) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_"),
         detection_type = if_else(detection_type == "acoustic", "aural", detection_type),
         date = format_iso_8601(date)) %>% 
  rename(duration_minutes = duration_min)

write_csv(penn_survey_table, here("clean_tables", "penn_survey.csv"))
  

```

## panama survey table
```{r}

panama_survey_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "survey.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  group_by(date, site, survey_time, detection_type) %>% 
  mutate(new_survey_id = cur_group_id(),
         detection_type = if_else(detection_type == "call", "aural", detection_type)) %>% 
  filter(!duplicated(new_survey_id)) %>% 
  select(!c(new_survey_id, e_dna)) %>% 
  rename(wind_speed_m_s = wind_speed_ms)

panama_survey_table <- panama_raw %>% 
  select(c((1:5), (6:15), (21:35), salinity_large, detection_type)) %>% 
  select(!c(survey_id, sort_id, merge_id, region, salinity_large, e_dna)) %>% 
  mutate(site = str_replace_all(site, " ", "_")) %>% 
  group_by(date, site, survey_time, detection_type) %>% 
  mutate(new_survey_id = cur_group_id()) %>% 
  filter(!duplicated(new_survey_id)) %>% 
  select(!c(new_survey_id)) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  plyr::rbind.fill(panama_survey_2022) %>% 
  rename(conductivity_uS_cm = conductivity_insertunits,
         salinity_ppt = salinity_small,
         tds_ppm = tds_insertunits) %>% 
  mutate(dissolved_o2_percent = as.numeric(dissolved_o2_percent), 
         salinity_ppt = if_else(salinity_ppt > "1", "", salinity_ppt),
         salinity_ppt = na_if(salinity_ppt, ""),
         dissolved_o2_percent = case_when(dissolved_o2_percent > 100 ~ 100,
                                          dissolved_o2_percent < 100 ~ dissolved_o2_percent),
         wind_speed_m_s = if_else(wind_speed_m_s == "23.7", "", wind_speed_m_s),
         wind_speed_m_s = na_if(wind_speed_m_s, ""),
        wind_speed_m_s = case_when(str_detect(wind_speed_m_s, "ph", negate = TRUE)~ wind_speed_m_s),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "/s", ""),
         wind_speed_m_s = str_replace_all(wind_speed_m_s, ", gust to 3.7", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "kt", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, "k/hr", ""),
        wind_speed_m_s = str_replace_all(wind_speed_m_s, " ", ""),
        transect = str_to_lower(str_replace_all(transect, " ", "_")),
        transect = str_replace_all(transect, ",", "_"),
        transect = str_replace_all(transect, "-", "_"))



write_csv(panama_survey_table, here("clean_tables", "panama_survey.csv"))
```

## read in serdp survey table
```{r}

serdp_survey_table <- read_csv(here("data", "serdp", "serdp_survey_table.csv")) %>% 
  mutate(survey_notes = str_to_lower(survey_notes),
         precipitation_last_48_h = str_to_lower(precipitation_last_48_h),
         survey_notes = str_replace_all(survey_notes, "[^[:alnum:]]", ""),
         weather_condition_notes = str_replace_all(weather_condition_notes, "[^[:alnum:]]", ""),
         site = str_to_lower(site),
         site = str_replace(site, "-", "_"),
         site = str_remove(site, "\\."),
         site = if_else(site == "wood lab", "wood lab pond", site),
         site = if_else(site == "rv", "rv pond", site),
         site = if_else(site == "tuttle", "tuttle pond", site),
         site = if_else(site == "phelps", "phelps pond", site),
         site = if_else(site == "newt (mike vorisek's)", "vorisek pond", site),
         site = str_replace_all(site, " ", "_"),
         survey_time = str_to_lower(survey_time),
         site = if_else(site_code == "la7", "la7", site),
         site = if_else(site_code == "la8", "la8", site),
         site = if_else(site == "la06", "la6", site),
         detection_type = "capture",
         date = format_iso_8601(date),
         duration_minutes = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_minutes = duration_minutes/60) %>% 
  unite("wind_speed_mpers_min", c("wind_speed_mpers_min", "wind_speed_min_mpers"), na.rm = T) %>% 
  unite("wind_speed_mpers_max", c("wind_speed_mpers_max", "wind_speed_max_mpers"), na.rm = T) %>% 
  unite("pressure_hg", c("pressure_in_hg", "pressure_in"), na.rm = T) %>% 
  select(!c(pressure_mb)) %>% 
  mutate(percent_vegetation_cover = if_else(percent_vegetation_cover == "huge lake", "", percent_vegetation_cover)) %>% 
  mutate_all(na_if, "")

serdp_survey_table$duration_minutes <- str_sub(serdp_survey_table$duration_minutes, -4) %>% 
  as.numeric()

write_csv(serdp_survey_table, here("clean_tables", "serdp_survey.csv"))
```

## sn survey table

### query data
```{sql, connection=connection, output.var = "sn_survey"}


select v.site_id as site, v.visit_date, s.* 
from visit v
join survey s on s.visit_id = v.id;


```

### clean sn survey table - need to figure out what to do about CMR and Swab
```{r}

sn_survey_table <- sn_survey %>%
  select(!c(id, visit_id)) %>% 
  rename(detection_type = survey_type,
         date = visit_date) %>% 
  mutate(survey_time = "day",
         detection_type = as.character(detection_type),
         detection_type = if_else(detection_type == "cmr", "capture", detection_type),
         detection_type = if_else(detection_type == "swab", "capture", detection_type)) %>% 
  group_by(site, date, survey_time, detection_type) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!c(temp_id))
  
write_csv(sn_survey_table, here("clean_tables", "sn_survey_table.csv"))

rm(sn_survey)
gc()
```


# create VES table

## penn1 ves table
```{r}

penn_ves <- penn1_VES %>% 
  select(!c(region, location, observer)) %>% 
  mutate(duration_min = if_else(end_time < start_time,
                            as_hms(86400) - start_time + end_time,
                            end_time - start_time),
         duration_min = duration_min/60,
         date = format_iso_8601(date)) %>% 
  relocate(duration_min, .before = species_ves) %>% 
  rename(ves_notes = comments_ves,
         duration_minutes = duration_min)

penn_ves$duration_minutes <- str_sub(penn_ves$duration_minutes, -4) %>% 
  as.numeric()


rm(penn1_VES)
gc()
```

## panama ves table 
```{r}

panama_ves_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "ves.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  rename(ves_notes = notes_aural)

panama_ves <- panama_raw %>% 
  filter(detection_type == "visual") %>% 
  select(site, date, survey_time, start_time, end_time, 
         duration_minutes, species, detection_type,
         quantity, capture_loc, microhab, life_stage, sex, notes_amphib1, 
         notes_amphib2, microhab0) %>% 
  unite(ves_notes, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         count = quantity,
         species_ves = species,
         detection_location = capture_loc) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  plyr::rbind.fill(panama_ves_2022)

  
```

## sn ves table

### query data
```{sql, connection=connection, output.var = "sn_ves"}


select v.site_id as site, v.visit_date as date, s.survey_type as detection_type, vs.*
from visit v
join survey s on v.id = s.visit_id
join visual_survey vs on s.id = vs.survey_id;


```

### clean sn ves data - need to align with other ves still
```{r}

sn_ves_table <- sn_ves %>% 
  select(!c(id, survey_id)) %>% 
  mutate(survey_time = "day") %>% 
  rename(species_ves = species,
         ves_notes = comment,
         microhab = location,
         life_stage = visual_life_stage)


#write_csv(sn_ves_table, here("clean_tables", "sn_ves_table.csv"))

rm(sn_ves)
gc()
```


## bind ves tables
```{r}

ves <- plyr::rbind.fill(panama_ves, penn_ves, sn_ves_table) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         sex = if_else(sex == "unkonwn", "unknown", sex)) %>% 
  rename(notes_ves = ves_notes) %>% 
  select(!c(start_time, end_time, duration_minutes)) %>% 
  mutate(count = if_else(count == "many", "100", count)) # need to find better fix to "many"

write_csv(ves, here("clean_tables", "ves.csv"))

rm(panama_ves, penn_ves, sn_ves_table)
gc()
```


# create aural table

## penn1 aural table
```{r}

penn_aural <- penn1_aural %>% 
  select(!c(region, observer, location, start_hour, end_hour, start_time, end_time, duration_min)) %>% 
  mutate(detection_type = "aural",
         species_acoustic = str_replace_all(species_acoustic, " ", "_"),
         site = str_replace_all(site, " ", "_"),
         site = str_replace(site, "-", "_")) %>% 
  rename(species_aural = species_acoustic,
         notes_aural = acoustic_comments)

#rm(penn1_aural)
gc()
```

## panama aural table
```{r}

panama_aural_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "aural.csv")) %>% 
  mutate(date = format_iso_8601(date))

panama_aural <- panama_raw %>% 
  filter(detection_type == "aural") %>% 
  select(site, date, survey_time, 
         species, detection_type,
         quantity, capture_loc, microhab, life_stage, sex, notes_amphib1, 
         notes_amphib2, microhab0) %>% 
  unite(notes_aural, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         count = quantity,
         species_aural = species,
         detection_location = capture_loc) %>% 
  mutate(date = format_iso_8601(date),
         microhab = if_else(is.na(microhab), "unknown", microhab),
         count = if_else(is.na(count), "1", count), #DOUBLE CHECK THIS IS CORRECT
         site = str_replace_all(site, " ", "_"),
         species_aural = str_replace_all(species_aural, " ", "_")) %>% 
  plyr::rbind.fill(panama_aural_2022)

```

## bind aural table
```{r}

aural <- plyr::rbind.fill(penn_aural, panama_aural) %>% 
  mutate(count = if_else(count == "many", "100", count),
         sex = if_else(sex == "unkonwn", "unknown", sex)) # need to find better fix for "many"

write_csv(aural, here("clean_tables", "aural.csv"))

#rm(penn_aural, panama_aural)
gc()
```

# Genomic data

## SERDP Legacy genomic - BD
need to add before capture table since serdp bd genomic data was not present in initial serdp data and need to back log genomic ids
## read in serdp bd genomic data
```{r}

raw_serdp_bd_genom <- read_csv(here("data", "serdp", "bd_genomic", "bd_genomic_serd.csv")) %>% 
  clean_names()

```


## pull out important columns from serdp bd genomic data for db
```{r}

serdp_bd_genom <- raw_serdp_bd_genom %>% 
  select(sample_id, seq_run, c(n_seqs_187:lineage_dapc_2_99cut)) %>% 
  rename(genetic_id = sample_id) %>% 
  filter(!genetic_id %in% c("180223-01C-LA02-LICA", "180223-04D-LA02-LICA", "180223-01F-LA02-LICA",
            "180223-04A-LA02-LICA", "180223-01B-LA02-LISP", "180223-01D-LA02-LISP"))

write_csv(serdp_bd_genom, here("clean_tables", "serdp_bd_genom.csv"))

# used for adding sample_id into serdp table
t <- serdp_bd_genom %>% 
  select(genetic_id) %>% 
  mutate(sample_id=genetic_id)


```

# Microbiome data

## Newt serdp data - contains ID for microbiome and mucosome and bacterial column...I think
need to add before capture table since serdp newt data is not accurate for binary microbiome column. Missing some data points
```{r}

newt <- read_csv(here("data", "serdp", "newt", "microbiome_antifungal_TTX_mucosome.csv")) %>% 
  mutate(mucosome_id = swab_id,
         microbiome_swab_id = swab_id)

newt <- newt[rowSums(is.na(newt)) != ncol(newt[,2:27]), ]

write_csv(newt, here("clean_tables", "serdp_newt_microbiome_mucosome_antifungal.csv"))

# use for back logging 62 missing IDs
all_newt_ids <- newt %>% 
  select(microbiome_swab_id, mucosome_id, swab_id)

```



# Capture table

## penn1 capture
```{r}

penn_capture <- penn1_capture %>% 
  select(!c(region, location, visit_comment, survey_comment, start_time, end_time,
            observer)) %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "-", "_"),
         species_capture = str_replace_all(species_capture, " ", "_"),
         bd_swab_tube_id = if_else(bd_swab_tube_id == "BdSwab00000", "NA", bd_swab_tube_id),
         dry_swab_tube_id=if_else(dry_swab_tube_id == "DrySwab00000", "NA", dry_swab_tube_id),
         crispr_swab_tube_id=if_else(crispr_swab_tube_id == "CRSwab00000", "NA", crispr_swab_tube_id),
         bacterial_swab_tube_id=if_else(bacterial_swab_tube_id == "BacSwab00000", "NA", bacterial_swab_tube_id),
         mucosome_id=if_else(mucusome_id == "MucBath00000", "NA", mucusome_id),
         amp_id_1=if_else(amp_id_1 == "AMPBath00000", "NA", amp_id_1),
         amp_id_2=if_else(amp_id_2 == "AMPBath00000", "NA", amp_id_2),
         amp_id_3=if_else(amp_id_3 == "AMPBath00000", "NA", amp_id_3),
         amp_id_4=if_else(amp_id_4 == "AMPBath00000", "NA", amp_id_4),
         antibody_id_1=if_else(antibody_id_1 == "AntiBod00000", "NA", antibody_id_1),
         antibody_id_2=if_else(antibody_id_2 == "AntiBod00000", "NA", antibody_id_2),
         antibody_id_3=if_else(antibody_id_3 == "AntiBod00000", "NA", antibody_id_3),
         antibody_id_4=if_else(antibody_id_4 == "AntiBod00000", "NA", antibody_id_4),
         toe_clip_tube_id=if_else(toe_clip_tube_id == "ToeClip00000", "NA", toe_clip_tube_id)) %>% 
  rename(bag_mass_g = bag_mass,
         svl_mm = snout_vent_length,
         body_and_bag_mass_g = body_and_bag_mass,
         body_mass_g = body_mass,
         substrate_temp_c = microhabitat_temperature_c,
         bacterial_swab_id =  bacterial_swab_tube_id, #glycerol/culture
         microbiome_swab_id = bd_swab_tube_id, #sequence/dry swab
         photo_id = bag_photo,
         genetic_id = toe_clip_tube_id,
         amp_id = amp_id_1,
         notes_capture = capture_comments,
         body_temp_c = body_temperature_c,
         bd_swab_id = dry_swab_tube_id,
         crispr_id = crispr_swab_tube_id,
         antibody_id = antibody_id_1) %>% 
  unite(life_stage, c(life_stage, life_stage_other), sep = ",", na.rm = T) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  select(!c(mucusome_id))
  

#rm(penn1_capture, penn1_VES, penn1_aural)
gc()
```

## sn capture

### queruy data
```{sql, connection=connection, output.var = "sn_capture"}


select v.site_id as site, v.visit_date as date, s.survey_type as detection_type, cs.*
from visit v
join survey s on v.id = s.visit_id
join capture_survey cs on s.id = cs.survey_id;


```

### clean sn_data 
```{r}

sn_capture_table <- sn_capture %>% 
  select(!c(survey_id, id, pit_tag_ref, tag_new)) %>% 
  mutate(survey_time = "day",
         detection_type = as.character(detection_type),
         detection_type = if_else(detection_type == "cmr", "capture", detection_type),
         detection_type = if_else(detection_type == "swab", "capture", detection_type)) %>% 
  rename(notes_capture = comment,
         body_mass_g = weight,
         svl_mm = length,
         life_stage = capture_life_stage,
         species_capture = species,
         microhabitat_type = location,
         processor = surveyor_id) %>% 
  mutate(microbiome_swab_id = if_else(str_detect(swab_id, "AJS"), swab_id, NULL), #metagenomic sequencing so micro
         bacterial_swab_id = if_else(str_detect(swab_id, "BII"), swab_id, NULL), #culture/glycerol
         mucosome_id = if_else(str_detect(swab_id, "BII"), swab_id, NULL)) %>% 
  rename(bd_swab_id = swab_id,
         capture_utme = utme,
         capture_utmn = utmn)

  
#write_csv(sn_capture_table, here("clean_tables", "sn_capture_table.csv"))
```


## brazil_legacy capture
```{r}
brazil_legacy_capture <- read_csv(here("data", "brazil_legacy", "brazil_capture.csv")) %>% 
  select(!c(17:20)) %>% 
  select(!c(count, family)) %>% 
  rename(species_capture = scientific_name,
         notes_capture = comments,
         bd_swab_id = swab_id,
         bacterial_swab_id = glycerol, #culture/glycerol
         genetic_id = tissue) %>% 
  unite(site, area:environment, sep = "_") %>% 
  mutate(detection_type = "capture",
         site = str_replace_all(site, " ", "_"),
         species_capture = str_replace_all(species_capture, " ", "_"),
         bacterial_swab_id = if_else(bacterial_swab_id == "TRUE", bd_swab_id, NULL),
         genetic_id = if_else(genetic_id == "TRUE", bd_swab_id, NULL),
         time_of_capture = if_else(time_of_capture == "NA:00", "", time_of_capture))

v <- vect(brazil_legacy_capture, c("capture_longitude", "capture_latitude"), crs="+proj=longlat")
u <- terra::project(v, "+proj=utm +zone=23")
utm <- crds(u)

i <- which(!is.na(brazil_legacy_capture$capture_longitude))
brazil_legacy_capture[i, c("capture_utme", "capture_utmn")] <- utm

brazil_legacy_capture <- brazil_legacy_capture %>% 
  select(!c(capture_longitude, capture_latitude))


# translation <- brazil_legacy_capture %>% 
#   select(notes_capture) %>% 
#   unique() %>% 
#   mutate(translation = "") %>% 
#   write_csv(here("legacy_brazil_translation.csv"))
```

## serdp capture table
```{r}

serdp_capture <- read_csv(here("data", "serdp", "serdp_capture_table.csv")) %>% 
  rename(bag_mass_g = mass_of_bag_g,
         time_of_capture = time,
         body_and_bag_mass_g = mass_of_animal_in_bag,
         body_mass_g = mass_of_animal_g,
         body_temp_c = body_temp,
         microhabitat_type = microhabitat_category) %>% 
  select(!c(species_code, frog_number, frog_id)) %>% 
  mutate(site = str_to_lower(site),
         site = str_replace_all(site, " ", "_"),
         bd_swab = if_else(bd_swab == 1, sample_id, NULL),
         mucosome = if_else(mucosome == 1, sample_id, NULL),
         bacterial_swab_id = if_else(microbiome == 1, sample_id, NULL), #culture/glycerol bd
         microbiome = if_else(microbiome == 1, sample_id, NULL), #sequce/dry swab
         am_ps = if_else(am_ps == 1, sample_id, NULL),
         antibodies = if_else(antibodies == 1, sample_id, NULL)) %>% 
  rename(bd_swab_id = bd_swab,
         mucosome_id = mucosome,
         amp_id = am_ps,
         microbiome_swab_id = microbiome, #micriobiome column both types, culture and sequence
         substrate_temp_c = substrate_temp,
         notes_capture = capture_comments,
         antibody_id = antibodies) %>% 
  left_join(t, by = c("sample_id")) %>% 
  select(!c(sample_id))


# back log missing 62 mucosome and microbiome ids
back_log_microb_muc <- serdp_capture %>%
  filter(!is.na(microbiome_swab_id)) %>% 
  select(microbiome_swab_id, mucosome_id, bd_swab_id) %>% 
  filter(bd_swab_id %in% all_newt_ids$swab_id)

Ids_to_back_log <- anti_join(all_newt_ids, back_log_microb_muc, by = c("swab_id" = "bd_swab_id"))

serdp_capture <- left_join(serdp_capture, Ids_to_back_log, by = c("bd_swab_id"="swab_id")) %>% 
  unite(microbiome_swab_id, c("microbiome_swab_id.x", "microbiome_swab_id.y"), na.rm = T) %>% 
  unite(mucosome_id, c("mucosome_id.x", "mucosome_id.y"), na.rm = T)


```

## panama capture table
```{r}

panama_capture_2022 <- read_csv(here("data", "panama_legacy", "clean_tables", "capture.csv")) %>% 
  mutate(date = format_iso_8601(date)) %>% 
  rename(capture_animal_state = dead,
         bd_swab_id = bdswab,
         mucosome_id = mucosome_water,
         bacterial_swab_id = glycerolswab, #culture bd
         genetic_id = oralswab,
         amp_id = amps,
         microbiome_swab_id = bacteriaswab) %>% # dry swab for sequencing 
  mutate(capture_animal_state = case_when(capture_animal_state == "1" ~ "dead",
                                          capture_animal_state == "0" ~ "alive"),
         bd_swab_id = if_else(bd_swab_id == "1", swab_id, NULL),
         mucosome_id = if_else(mucosome_id == "1", swab_id, NULL),
         bacterial_swab_id = if_else(bacterial_swab_id == "1", swab_id, NULL), #culture/glycerol 
         genetic_id = if_else(genetic_id == "1", swab_id, NULL),
         amp_id = if_else(amp_id == "1", swab_id, NULL),
         ampshcl = if_else(ampshcl == "1", ampshcl, NULL),
         microbiome_swab_id = if_else(microbiome_swab_id == "1", swab_id, NULL)) %>% 
  select(!c(swab_id, count))


# panama data fixes
panama_rabbit_stream_fix <- panama_raw %>% 
  filter(site == "rabbit stream",
         date == "2014-05-27") %>% 
  group_by(swab_id) %>% 
  mutate(temp_id = cur_group_id()) %>% 
  filter(!duplicated(temp_id)) %>% 
  select(!temp_id)


panama_capture <- panama_raw %>%
  filter(!swab_id %in% panama_rabbit_stream_fix$swab_id) %>% 
  rbind(panama_rabbit_stream_fix) %>% 
  filter(detection_type == "capture") %>% 
  select(c(site, date, survey_time, swab_id, species, detection_type, capture_time,
           capture_loc, microhab, body_temp, subs_temp, svl, frog_mass, life_stage, sex, dead, #infected, pcr,
           bdswab, gen_samp, bacteria_swab, am_ps, e_dna, photo, photo_id, notes_amphib1,
           notes_amphib2, microhab0)) %>% 
  unite(notes_capture, notes_amphib1:notes_amphib2, sep = ", ", na.rm = T) %>% 
  rename(microhab_moredetail = microhab0,
         time_of_capture = capture_time,
         capture_trx_loc = capture_loc,
         microhabitat_type = microhab,
         body_temp_c = body_temp,
         substrate_temp_c = subs_temp,
         svl_mm = svl,
         body_mass_g = frog_mass) %>% 
  mutate(bdswab = if_else(bdswab=="1", swab_id, NULL),
         am_ps = if_else(am_ps=="1", swab_id, NULL),
         bacteria_swab = if_else(bacteria_swab=="1", swab_id, NULL),
         e_dna = if_else(e_dna=="1", swab_id, NULL),
         gen_samp = case_when(gen_samp == "toeclip" ~ swab_id,
                              gen_samp == "buccal" ~ swab_id,
                              gen_samp == "1" ~ swab_id),
         site = str_replace_all(site, " ", "_"),
         species = str_replace_all(species, " ", "_"),
         date = format_iso_8601(date)) %>% 
  rename(amp_id = am_ps,
         genetic_id = gen_samp,
         bd_swab_id = bdswab,
         microbiome_swab_id = bacteria_swab, #dry swab frozen
         species_capture = species,
         capture_animal_state = dead) %>% 
  mutate(capture_animal_state = case_when(capture_animal_state == "1" ~ "dead",
                                          capture_animal_state == "0" ~ "alive")) %>% 
  select(!c(swab_id)) %>% 
  plyr::rbind.fill(panama_capture_2022) 
  


```

## bind all capture tables
```{r}

capture <- plyr::rbind.fill(panama_capture, brazil_legacy_capture, penn_capture, serdp_capture,
                            sn_capture_table) %>% 
  mutate(notes_capture = str_replace_all(notes_capture, "[[:punct:]]", ""),
         site = str_replace_all(site, "-", "_"),
         site = str_replace_all(site, " ", "_"),
         microhabitat_notes = str_replace_all(microhabitat_notes, "[[:punct:]]", ""),
         site = if_else(site == "wood_lab", "wood_lab_pond", site),
         detection_type = "capture") %>% 
  mutate_all(na_if, "")

write_csv(capture, here("clean_tables", "capture.csv"))

# dups <- capture %>% 
#   group_by(bd_swab_id) %>% 
#   mutate(temp_id = cur_group_id()) %>% 
#   filter(duplicated(temp_id)) %>% 
#   drop_na(bd_swab_id)
# 
# z <- capture %>% 
#   filter(bd_swab_id == "32") # one is control

rm(panama_capture, brazil_legacy_capture, penn_capture, serdp_capture)
gc()
```

# AMP data

## SERPD AMP - contains ID for amp column and antibody column
```{r}

serdp_amp <- read_csv(here("data", "serdp", "amp", "SERDP_AMP.csv"))

```


# legacy hobo data

## hobo site table
```{r}

hobo_site <- serdp_sites  %>% 
  mutate(site = str_replace_all(site, " ", "_"),
         site = str_replace_all(site, "-", "_"))

write_csv(hobo_site, here("clean_tables", "hobo_site.csv"))
```

## read in serdp hobo data

### shade hobo
```{r}

raw_shade_hobo <- read.csv(here("data", "serdp", "hobo", "Shade_HOBO_2021_01_28.csv")) %>% 
  clean_names() 

shade_hobo <- raw_shade_hobo %>% 
  select(!c(x_1, x)) %>% 
  rename(shade_temperature_c = temperature,
         relative_humidity = rh,
         site_code = site,
         enviro_location = location,
         dew_point_c = dew_point) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))),
         dew_point_c = if_else(dew_point_c == "Logged", "", dew_point_c)) %>% 
  mutate_all(na_if, "")

write_csv(shade_hobo, here("clean_tables", "shade_hobo.csv"))

rm(raw_shade_hobo)
gc()
```

### soil hobo
```{r}

raw_soil_hobo <- read_csv(here("data", "serdp", "hobo", "Soil_HOBO_2021_01_28.csv")) %>% 
  clean_names()

soil_hobo <- raw_soil_hobo %>% 
  select(!c(x1, x)) %>% 
  rename(soil_temperature_c = temperature,
         site_code = site,
         enviro_location = location) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))))
write_csv(soil_hobo, here("clean_tables", "soil_hobo.csv"))

rm(raw_soil_hobo)
gc()
```

### sun hobo
```{r}

raw_sun_hobo <- read_csv(here("data", "serdp", "hobo", "Sun_HOBO_2021_01_28.csv")) %>% 
  clean_names()

sun_hobo <- raw_sun_hobo %>% 
  select(!c(x1, x)) %>% 
  rename(sun_temperature_c = temperature,
         site_code = site,
         enviro_location = location) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))))

write_csv(sun_hobo, here("clean_tables", "sun_hobo.csv"))

rm(raw_sun_hobo)
gc()
```

### water hobo

```{r}
raw_water_hobo <- read_csv(here("data", "serdp", "hobo", "Water_HOBO_2021_01_28.csv")) %>% 
  clean_names()

water_hobo <- raw_water_hobo %>% 
  select(!c(x1, x)) %>% 
  rename(water_temperature_c = temperature,
         site_code = site,
         enviro_location = location) %>% 
  mutate(site_code = str_to_lower(str_replace_all(site_code, "0", "")),
         date_time = format_iso_8601(parse_date_time(date_time, c("mdy H", "mdy HM", 
                                                                  "mdy HMS", "mdy"))),
         light = if_else(light == "Logged", "", light))%>% 
  mutate_all(na_if, "")

write_csv(water_hobo, here("clean_tables", "water_hobo.csv"))

rm(raw_water_hobo)
gc()
```


